// @inkline/inkline/components/ISidebar/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ISidebarBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ISidebarCollapseBreakpointExample", {
  enumerable: true,
  get: function () {
    return _collapseBreakpoint.default;
  }
});
Object.defineProperty(exports, "ISidebarCollapseFalseExample", {
  enumerable: true,
  get: function () {
    return _collapseFalse.default;
  }
});
Object.defineProperty(exports, "ISidebarCollapsePositionExample", {
  enumerable: true,
  get: function () {
    return _collapsePosition.default;
  }
});
Object.defineProperty(exports, "ISidebarCollapseTrueExample", {
  enumerable: true,
  get: function () {
    return _collapseTrue.default;
  }
});
Object.defineProperty(exports, "ISidebarCollapsibleExample", {
  enumerable: true,
  get: function () {
    return _collapsible.default;
  }
});
Object.defineProperty(exports, "ISidebarColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "ISidebarPlacementExample", {
  enumerable: true,
  get: function () {
    return _placement.default;
  }
});
Object.defineProperty(exports, "ISidebarRoutingActiveExample", {
  enumerable: true,
  get: function () {
    return _routingActive.default;
  }
});
Object.defineProperty(exports, "ISidebarRoutingExample", {
  enumerable: true,
  get: function () {
    return _routing.default;
  }
});
Object.defineProperty(exports, "ISidebarSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("./basic.vue"));
var _collapseBreakpoint = _interopRequireDefault(require("./collapse-breakpoint.vue"));
var _collapseTrue = _interopRequireDefault(require("./collapse-true.vue"));
var _collapseFalse = _interopRequireDefault(require("./collapse-false.vue"));
var _colorVariants = _interopRequireDefault(require("./color-variants.vue"));
var _collapsePosition = _interopRequireDefault(require("./collapse-position.vue"));
var _collapsible = _interopRequireDefault(require("./collapsible.vue"));
var _sizeVariants = _interopRequireDefault(require("./size-variants.vue"));
var _placement = _interopRequireDefault(require("./placement.vue"));
var _routing = _interopRequireDefault(require("./routing.vue"));
var _routingActive = _interopRequireDefault(require("./routing-active.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
