// @inkline/inkline/components/IHeader/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IHeaderBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IHeaderColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "IHeaderCoverExample", {
  enumerable: true,
  get: function () {
    return _cover.default;
  }
});
Object.defineProperty(exports, "IHeaderCoverExampleCSS", {
  enumerable: true,
  get: function () {
    return _cover2.default;
  }
});
Object.defineProperty(exports, "IHeaderFullscreenExample", {
  enumerable: true,
  get: function () {
    return _fullscreen.default;
  }
});
Object.defineProperty(exports, "IHeaderSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("./basic.vue"));
var _colorVariants = _interopRequireDefault(require("./color-variants.vue"));
var _cover = _interopRequireDefault(require("./cover.vue"));
var _cover2 = _interopRequireDefault(require("./cover.scss"));
var _fullscreen = _interopRequireDefault(require("./fullscreen.vue"));
var _sizeVariants = _interopRequireDefault(require("./size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
