// @inkline/inkline/components/IBreadcrumb/script.mjs
import { defineComponent } from "vue";
import {
  colorVariantClass,
  defaultPropValue,
  sizePropValidator
} from "@inkline/inkline/mixins";
const componentName = "IBreadcrumb";
export default defineComponent({
  name: componentName,
  inheritAttrs: false,
  props: {
    /**
     * The aria-label of the breadcrumbs
     * @type String
     * @default Breadcrumbs
     * @name ariaLabel
     */
    ariaLabel: {
      type: String,
      default: "Breadcrumbs"
    },
    /**
     * The color variant of the breadcrumb
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: defaultPropValue(componentName, "color")
    },
    /**
     * The size variant of the breadcrumb
     * @type sm | md | lg
     * @default md
     * @name size
     */
    size: {
      type: String,
      default: defaultPropValue(componentName, "size"),
      validator: sizePropValidator
    }
  },
  computed: {
    classes() {
      return {
        ...colorVariantClass(this),
        [`-${this.size}`]: Boolean(this.size)
      };
    }
  }
});
