// @inkline/inkline/components/ITooltip/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ITooltipBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ITooltipColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "ITooltipFreeformExample", {
  enumerable: true,
  get: function () {
    return _freeform.default;
  }
});
Object.defineProperty(exports, "ITooltipPlacementExample", {
  enumerable: true,
  get: function () {
    return _placement.default;
  }
});
Object.defineProperty(exports, "ITooltipSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
Object.defineProperty(exports, "ITooltipTriggerExample", {
  enumerable: true,
  get: function () {
    return _trigger.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/basic.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/color-variants.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/size-variants.vue"));
var _freeform = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/freeform.vue"));
var _placement = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/placement.vue"));
var _trigger = _interopRequireDefault(require("@inkline/inkline/components/ITooltip/examples/trigger.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
