// @inkline/inkline/components/IButtonGroup/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
const componentName = "IButtonGroup";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  inject: {
    form: {
      default: () => ({})
    },
    buttonGroup: {
      default: () => ({})
    },
    formGroup: {
      default: () => ({})
    }
  },
  provide() {
    return {
      buttonGroup: this
    };
  },
  inheritAttrs: false,
  props: {
    /**
     * Display the button group with vertical orientation
     * @type Boolean
     * @default false
     * @name vertical
     */
    vertical: {
      type: Boolean,
      default: false
    },
    /**
     * Display the button group as a block, spanning the full container width
     * @type Boolean
     * @default false
     * @name block
     */
    block: {
      type: Boolean,
      default: false
    },
    /**
     * The disabled state of the button group
     * @type Boolean
     * @default false
     * @name disabled
     */
    disabled: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classes() {
      return {
        "-vertical": this.vertical,
        "-block": this.block,
        "-disabled": this.isDisabled
      };
    },
    isDisabled() {
      return this.disabled || this.buttonGroup.disabled || this.form.disabled || this.formGroup.disabled;
    }
  }
});
module.exports = _default;
