// @inkline/inkline/components/ICard/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ICardBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ICardBodyExample", {
  enumerable: true,
  get: function () {
    return _cardBody.default;
  }
});
Object.defineProperty(exports, "ICardColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "ICardHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _cardHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ICardImageExample", {
  enumerable: true,
  get: function () {
    return _image.default;
  }
});
Object.defineProperty(exports, "ICardSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/basic.vue"));
var _cardBody = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/card-body.vue"));
var _cardHeaderFooter = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/card-header-footer.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/color-variants.vue"));
var _image = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/image.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/ICard/examples/size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
