// @inkline/inkline/components/ITable/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ITableBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ITableBorderedExample", {
  enumerable: true,
  get: function () {
    return _bordered.default;
  }
});
Object.defineProperty(exports, "ITableColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "ITableHoverExample", {
  enumerable: true,
  get: function () {
    return _hover.default;
  }
});
Object.defineProperty(exports, "ITableResponsiveExample", {
  enumerable: true,
  get: function () {
    return _responsive.default;
  }
});
Object.defineProperty(exports, "ITableStripedExample", {
  enumerable: true,
  get: function () {
    return _striped.default;
  }
});
var _basic = _interopRequireDefault(require("./basic.vue"));
var _bordered = _interopRequireDefault(require("./bordered.vue"));
var _colorVariants = _interopRequireDefault(require("./color-variants.vue"));
var _hover = _interopRequireDefault(require("./hover.vue"));
var _responsive = _interopRequireDefault(require("./responsive.vue"));
var _striped = _interopRequireDefault(require("./striped.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
