// @inkline/inkline/components/IListGroup/components/IListGroupItem/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "IListGroupItem";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  mixins: [_mixins.LinkableMixin],
  inheritAttrs: false,
  props: {
    /**
     * The active state of the list group item
     * @type Boolean
     * @default false
     * @name active
     */
    active: {
      type: Boolean,
      default: false
    },
    /**
     * The disabled state of the list group item
     * @type Boolean
     * @default false
     * @name disabled
     */
    disabled: {
      type: Boolean,
      default: false
    },
    /**
     * Set the HTML tag to be used for rendering the list group item
     * @type String
     * @default div
     * @name tag
     */
    tag: {
      type: String,
      default: "div"
    },
    /**
     * The tabindex of the list group item
     * @type Number | String
     * @default 0
     * @name tabindex
     */
    tabindex: {
      type: [Number, String],
      default: 0
    }
  },
  computed: {
    ariaDisabled() {
      if (this.role === "link") {
        return null;
      }
      return this.disabled ? "true" : "false";
    },
    classes() {
      return {
        "-active": this.active,
        "-disabled": this.disabled
      };
    },
    role() {
      return this.$attrs.to || this.$attrs.href ? "link" : "listitem";
    },
    tabIndex() {
      return this.disabled ? -1 : this.tabindex;
    }
  }
});
module.exports = _default;
