// @inkline/inkline/components/IBreadcrumb/components/IBreadcrumbItem/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "IBreadcrumbItem";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  mixins: [_mixins.LinkableMixin],
  inheritAttrs: false,
  props: {
    /**
     * The active state of the breadcrumb item
     * @type Boolean
     * @default false
     * @name active
     */
    active: {
      type: Boolean,
      default: false
    },
    /**
     * The disabled state of the breadcrumb item
     * @type Boolean
     * @default false
     * @name disabled
     */
    disabled: {
      type: Boolean,
      default: false
    },
    /**
     * The href property of the breadcrumb item rendered as anchor
     * @type string
     * @default undefined
     * @name href
     */
    href: {
      type: String,
      default: ""
    },
    /**
     * The to property of the breadcrumb item rendered as router link component
     * @type string | object
     * @default undefined
     * @name to
     */
    to: {
      type: [String, Object],
      default: ""
    },
    /**
     * The tabindex of the breadcrumb item
     * @type Number | String
     * @default 0
     * @name tabindex
     */
    tabindex: {
      type: [Number, String],
      default: 0
    }
  },
  computed: {
    classes() {
      return {
        "-active": this.active,
        "-disabled": this.disabled
      };
    },
    tabIndex() {
      return this.disabled || this.active ? -1 : this.tabindex;
    }
  }
});
module.exports = _default;
