// @inkline/inkline/components/ICollapsible/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ICollapsibleAccordionExample", {
  enumerable: true,
  get: function () {
    return _accordion.default;
  }
});
Object.defineProperty(exports, "ICollapsibleBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ICollapsibleColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "ICollapsibleDefaultOpenExample", {
  enumerable: true,
  get: function () {
    return _defaultOpen.default;
  }
});
Object.defineProperty(exports, "ICollapsibleHeaderExample", {
  enumerable: true,
  get: function () {
    return _header.default;
  }
});
Object.defineProperty(exports, "ICollapsibleSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _accordion = _interopRequireDefault(require("./accordion.vue"));
var _basic = _interopRequireDefault(require("./basic.vue"));
var _colorVariants = _interopRequireDefault(require("./color-variants.vue"));
var _defaultOpen = _interopRequireDefault(require("./default-open.vue"));
var _header = _interopRequireDefault(require("./header.vue"));
var _sizeVariants = _interopRequireDefault(require("./size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
