// @inkline/inkline/components/IProgress/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "IProgress";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  provide() {
    return {
      progress: this
    };
  },
  inheritAttrs: false,
  props: {
    /**
     * The color variant of the progress component
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "color")
    },
    /**
     * The value to consider as the 0% starting point
     * @type Number
     * @default 0
     * @name min
     */
    min: {
      type: [String, Number],
      default: 0
    },
    /**
     * The value to consider as the 100% ending point
     * @type Number
     * @default 100
     * @name max
     */
    max: {
      type: [String, Number],
      default: 100
    },
    /**
     * The size variant of the progress component
     * @type sm | md | lg
     * @default md
     * @name size
     */
    size: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "size"),
      validator: _mixins.sizePropValidator
    }
  },
  computed: {
    classes() {
      return {
        ...(0, _mixins.colorVariantClass)(this),
        [`-${this.size}`]: Boolean(this.size)
      };
    }
  }
});
module.exports = _default;
