// @inkline/inkline/components/ICollapsible/components/ICollapsibleItem/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _helpers = require("@inkline/inkline/helpers");
var _transitions = require("@inkline/inkline/transitions");
const componentName = "ICollapsibleItem";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  components: {
    IExpandTransition: _transitions.IExpandTransition
  },
  inject: {
    collapsible: {
      default: () => ({
        activeItems: []
      })
    }
  },
  inheritAttrs: false,
  props: {
    /**
     * The unique identifier of the collapsible item, used for determining if the item is open or not
     * @type String
     * @default uid()
     * @name name
     */
    name: {
      type: String,
      default() {
        return (0, _helpers.uid)("collapsible-item");
      }
    },
    /**
     * The title of the collapsible item
     * @type String
     * @default
     * @name title
     */
    title: {
      type: String,
      default: ""
    }
  },
  computed: {
    active() {
      return this.collapsible.activeItems.indexOf(this.name) > -1;
    },
    classes() {
      return {
        "-active": this.active
      };
    }
  },
  methods: {
    onClick() {
      this.collapsible.onItemClick(this);
    }
  }
});
module.exports = _default;
