// @inkline/inkline/components/IPagination/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IPaginationBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IPaginationColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "IPaginationLimitExample", {
  enumerable: true,
  get: function () {
    return _limit.default;
  }
});
Object.defineProperty(exports, "IPaginationLimitResponsiveExample", {
  enumerable: true,
  get: function () {
    return _limitResponsive.default;
  }
});
Object.defineProperty(exports, "IPaginationQuickLinksExample", {
  enumerable: true,
  get: function () {
    return _quickLinks.default;
  }
});
Object.defineProperty(exports, "IPaginationSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/basic.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/color-variants.vue"));
var _limit = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/limit.vue"));
var _limitResponsive = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/limit-responsive.vue"));
var _quickLinks = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/quick-links.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/IPagination/examples/size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
