// @inkline/inkline/components/IColumn/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.properties = exports.default = void 0;
var _vue = require("vue");
var _constants = require("@inkline/inkline/constants");
var _helpers = require("@inkline/inkline/helpers");
const properties = {};
exports.properties = properties;
for (const breakpoint of _constants.breakpointKeys) {
  if (breakpoint !== "") {
    properties[breakpoint] = {
      type: [String, Boolean, Number],
      default: false
    };
  }
  for (const property of ["first", "last"]) {
    properties[`${property}${(0, _helpers.capitalizeFirst)(breakpoint)}`] = {
      type: Boolean,
      default: false
    };
  }
  for (const property of ["offset", "push", "pull"]) {
    properties[`${property}${(0, _helpers.capitalizeFirst)(breakpoint)}`] = {
      type: [String, Number],
      default: ""
    };
  }
}
const componentName = "IColumn";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  inheritAttrs: false,
  props: properties,
  computed: {
    classes() {
      return Object.keys(properties).reduce((acc, property) => {
        if (this[property]) {
          acc[(0, _helpers.breakpointClass)(`-${property}`, this[property])] = true;
        }
        return acc;
      }, {});
    }
  }
});
module.exports = _default;
