// @inkline/inkline/components/IRadio/components/IRadioGroup/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
var _helpers = require("@inkline/inkline/helpers");
const componentName = "IRadioGroup";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  mixins: [_mixins.FormComponentMixin],
  provide() {
    return {
      formGroup: this
    };
  },
  inheritAttrs: false,
  props: {
    /**
     * The color variant of the radio group
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "color")
    },
    /**
     * The disabled state of the radio group
     * @type Boolean
     * @default false
     * @name disabled
     */
    disabled: {
      type: Boolean,
      default: false
    },
    /**
     * Display the radio group as inline
     * @type Boolean
     * @default false
     * @name inline
     */
    inline: {
      type: Boolean,
      default: false
    },
    /**
     * Used to set the radio group value
     * @default ''
     * @name modelValue
     */
    modelValue: {
      default: ""
    },
    /**
     * The unique identifier of the radio group
     * @type String
     * @default uid()
     * @name name
     */
    name: {
      type: [String, Number],
      default() {
        return (0, _helpers.uid)("radio-group");
      }
    },
    /**
     * The readonly state of the radio group
     * @type Boolean
     * @default false
     * @name readonly
     */
    readonly: {
      type: Boolean,
      default: false
    },
    /**
     * The size variant of the radio group
     * @type sm | md | lg
     * @default md
     * @name size
     */
    size: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "size"),
      validator: _mixins.sizePropValidator
    }
  },
  emits: [
  /**
   * Event emitted for setting the modelValue
   * @event update:modelValue
   */
  "update:modelValue"],
  computed: {
    classes() {
      return {
        ...(0, _mixins.colorVariantClass)(this),
        [`-${this.size}`]: Boolean(this.size),
        "-disabled": this.isDisabled,
        "-readonly": this.isReadonly,
        "-inline": this.inline
      };
    },
    checked() {
      if (this.schema) {
        return this.schema.value;
      }
      return this.modelValue;
    }
  },
  methods: {
    onChange(value) {
      this.parent.onInput?.(this.name, value);
      this.$emit("update:modelValue", value);
    }
  }
});
module.exports = _default;
