// @inkline/inkline/components/ITabs/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "ITabs";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  provide() {
    return {
      tabs: this
    };
  },
  inheritAttrs: false,
  props: {
    /**
     * The color variant of the tabs
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "color")
    },
    /**
     * Used to set the currently active tab
     * @type String
     * @default
     * @name modelValue
     */
    modelValue: {
      type: String,
      default: ""
    },
    /**
     * The size variant of the tabs
     * @type sm | md | lg
     * @default md
     * @name size
     */
    size: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "size"),
      validator: _mixins.sizePropValidator
    },
    /**
     * Display the tabs header as full width
     * @type Boolean
     * @default false
     * @name stretch
     */
    stretch: {
      type: Boolean,
      default: false
    }
  },
  emits: [
  /**
   * Event emitted for setting the modelValue
   * @event update:modelValue
   */
  "update:modelValue"],
  data() {
    return {
      active: this.modelValue,
      tabs: []
    };
  },
  computed: {
    classes() {
      return {
        ...(0, _mixins.colorVariantClass)(this),
        [`-${this.size}`]: Boolean(this.size),
        "-stretch": this.stretch
      };
    }
  },
  watch: {
    modelValue(value) {
      this.active = value;
    }
  },
  methods: {
    setActive(id) {
      this.active = id;
      this.$emit("update:modelValue", this.active);
    }
  }
});
module.exports = _default;
