// @inkline/inkline/validation/validators/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "alpha", {
  enumerable: true,
  get: function () {
    return _alpha.alpha;
  }
});
Object.defineProperty(exports, "alphanumeric", {
  enumerable: true,
  get: function () {
    return _alphanumeric.alphanumeric;
  }
});
Object.defineProperty(exports, "custom", {
  enumerable: true,
  get: function () {
    return _custom.custom;
  }
});
Object.defineProperty(exports, "email", {
  enumerable: true,
  get: function () {
    return _email.email;
  }
});
Object.defineProperty(exports, "max", {
  enumerable: true,
  get: function () {
    return _max.max;
  }
});
Object.defineProperty(exports, "maxLength", {
  enumerable: true,
  get: function () {
    return _maxLength.maxLength;
  }
});
Object.defineProperty(exports, "min", {
  enumerable: true,
  get: function () {
    return _min.min;
  }
});
Object.defineProperty(exports, "minLength", {
  enumerable: true,
  get: function () {
    return _minLength.minLength;
  }
});
Object.defineProperty(exports, "number", {
  enumerable: true,
  get: function () {
    return _number.number;
  }
});
exports.registerValidator = registerValidator;
Object.defineProperty(exports, "required", {
  enumerable: true,
  get: function () {
    return _required.required;
  }
});
Object.defineProperty(exports, "sameAs", {
  enumerable: true,
  get: function () {
    return _sameAs.sameAs;
  }
});
exports.unregisterValidator = unregisterValidator;
exports.validators = void 0;
var _alpha = require("@inkline/inkline/validation/validators/alpha");
var _alphanumeric = require("@inkline/inkline/validation/validators/alphanumeric");
var _custom = require("@inkline/inkline/validation/validators/custom");
var _number = require("@inkline/inkline/validation/validators/number");
var _email = require("@inkline/inkline/validation/validators/email");
var _max = require("@inkline/inkline/validation/validators/max");
var _maxLength = require("@inkline/inkline/validation/validators/maxLength");
var _min = require("@inkline/inkline/validation/validators/min");
var _minLength = require("@inkline/inkline/validation/validators/minLength");
var _required = require("@inkline/inkline/validation/validators/required");
var _sameAs = require("@inkline/inkline/validation/validators/sameAs");
const validators = {
  alpha: _alpha.alpha,
  alphanumeric: _alphanumeric.alphanumeric,
  custom: _custom.custom,
  number: _number.number,
  email: _email.email,
  max: _max.max,
  maxLength: _maxLength.maxLength,
  min: _min.min,
  minLength: _minLength.minLength,
  required: _required.required,
  sameAs: _sameAs.sameAs
};
exports.validators = validators;
function registerValidator(name, validator) {
  validators[name] = validator;
}
function unregisterValidator(name) {
  delete validators[name];
}
