// @inkline/inkline/validation/validators/index.mjs
import { alpha } from "./alpha.mjs";
import { alphanumeric } from "./alphanumeric.mjs";
import { custom } from "./custom.mjs";
import { number } from "./number.mjs";
import { email } from "./email.mjs";
import { max } from "./max.mjs";
import { maxLength } from "./maxLength.mjs";
import { min } from "./min.mjs";
import { minLength } from "./minLength.mjs";
import { required } from "./required.mjs";
import { sameAs } from "./sameAs.mjs";
export { alpha } from "./alpha.mjs";
export { alphanumeric } from "./alphanumeric.mjs";
export { custom } from "./custom.mjs";
export { number } from "./number.mjs";
export { email } from "./email.mjs";
export { max } from "./max.mjs";
export { maxLength } from "./maxLength.mjs";
export { min } from "./min.mjs";
export { minLength } from "./minLength.mjs";
export { required } from "./required.mjs";
export { sameAs } from "./sameAs.mjs";
export const validators = {
  alpha,
  alphanumeric,
  custom,
  number,
  email,
  max,
  maxLength,
  min,
  minLength,
  required,
  sameAs
};
export function registerValidator(name, validator) {
  validators[name] = validator;
}
export function unregisterValidator(name) {
  delete validators[name];
}
