// @inkline/inkline/__mocks__/useServer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useServer = exports.usePagination = void 0;
var _miragejs = require("miragejs");
var _data = _interopRequireDefault(require("./data.json"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const usePagination = (schema, req, model) => {
  const {
    models: items
  } = schema[model].all();
  const page = req.queryParams.page ? parseInt(req.queryParams.page, 10) : 1;
  const rowsPerPage = req.queryParams.rowsPerPage ? parseInt(req.queryParams.rowsPerPage, 10) : 25;
  const query = req.queryParams.query;
  const filteredItems = query ? items.filter(item => `${item.firstName} ${item.lastName}`.toLowerCase().includes(query.toLowerCase())) : items;
  return {
    items: filteredItems.slice((page - 1) * rowsPerPage, page * rowsPerPage),
    total: filteredItems.length
  };
};
exports.usePagination = usePagination;
let server;
const useServer = () => {
  if (!server) {
    server = (0, _miragejs.createServer)({
      models: {
        user: _miragejs.Model
      },
      seeds(server2) {
        _data.default.forEach(row => server2.schema.users.create(row));
      },
      routes() {
        this.namespace = "api";
        this.get("/users", (schema, req) => usePagination(schema, req, "users"));
      }
    });
  }
  return server;
};
exports.useServer = useServer;
