// tailvue/dist/tailvue.es.js
import { defineComponent as E, computed as F, openBlock as h, createElementBlock as y, normalizeClass as S, unref as p, renderSlot as Re, createElementVNode as d, createCommentVNode as v, getCurrentScope as ut, onScopeDispose as ft, watch as dt, h as gt, ref as A, onMounted as Ie, createVNode as w, Transition as me, withCtx as R, createApp as ht, createBlock as C, toDisplayString as _, createTextVNode as Ae, normalizeStyle as X, getCurrentInstance as mt } from "vue";
const se = [
  {
    name: "white",
    primary: "border border-gray-300 shadow-sm font-medium rounded text-gray-700 bg-white",
    dark: "dark:border-gray-600 dark:bg-gray-700 dark:text-gray-400 hover:dark:bg-gray-600 dark:ring-offset-gray-800",
    active: "hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500",
    disabled: ""
  },
  {
    name: "dark",
    primary: "border border-gray-600 shadow-sm font-medium rounded text-gray-400 bg-gray-700",
    active: "hover:bg-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 ring-offset-gray-800",
    disabled: ""
  },
  {
    name: "white-left",
    primary: "border border-gray-300 shadow-sm font-medium rounded-l-md text-gray-700 bg-white",
    dark: "dark:border-gray-600 dark:bg-gray-700 dark:text-gray-400 hover:dark:bg-gray-500 dark:ring-offset-gray-800",
    active: "hover:bg-gray-50 focus:z-10 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500",
    disabled: ""
  },
  {
    name: "white-mid",
    primary: "-ml-px relative inline-flex items-center border border-gray-300 shadow-sm font-medium text-gray-700 bg-white",
    dark: "dark:border-gray-600 dark:bg-gray-700 dark:text-gray-400 hover:dark:bg-gray-500 dark:ring-offset-gray-800",
    active: "hover:bg-gray-50 focus:z-10 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500",
    disabled: ""
  },
  {
    name: "white-right",
    primary: "-ml-px border border-gray-300 shadow-sm font-medium rounded-r-md text-gray-700 bg-white",
    dark: "dark:border-gray-600 dark:bg-gray-700 dark:text-gray-400 hover:dark:bg-gray-500 dark:ring-offset-gray-800",
    active: "hover:bg-gray-50 focus:z-10 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500",
    disabled: ""
  },
  {
    name: "text",
    primary: "text-gray-700",
    active: "hover:text-gray-500 hover:bg-gray-100 focus:outline-none active:text-gray-800 active:bg-gray-50"
  },
  {
    name: "indigo",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-indigo-600",
    active: "hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500",
    disabled: ""
  },
  {
    name: "indigo-light",
    primary: "text-indigo-700 bg-indigo-100",
    active: "hover:bg-indigo-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 active:bg-indigo-200",
    disabled: ""
  },
  {
    name: "indigo-dark",
    primary: "text-indigo-200 bg-indigo-900 rounded shadow-sm ",
    active: "hover:bg-indigo-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-300 active:bg-indigo-800",
    disabled: ""
  },
  {
    name: "red",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-red-600",
    active: "hover:bg-red-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 focus:shadow-outline-red active:bg-red-700",
    disabled: ""
  },
  {
    name: "yellow",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-yellow-600",
    active: "hover:bg-yellow-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-yellow-500 focus:shadow-outline-yellow active:bg-yellow-700",
    disabled: ""
  },
  {
    name: "green",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-green-600",
    active: "hover:bg-green-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500 focus:shadow-outline-green active:bg-green-700",
    disabled: ""
  },
  {
    name: "emerald",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-emerald-600",
    active: "hover:bg-emerald-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-emerald-500 focus:shadow-outline-emerald active:bg-emerald-700",
    disabled: ""
  },
  {
    name: "teal",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-teal-600",
    active: "hover:bg-teal-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500 focus:shadow-outline-teal active:bg-teal-700",
    disabled: ""
  },
  {
    name: "blue",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-blue-600",
    active: "hover:bg-blue-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 focus:shadow-outline-blue active:bg-blue-700",
    disabled: ""
  },
  {
    name: "light-blue",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-light-blue-600",
    active: "hover:bg-light-blue-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-light-blue-500 focus:shadow-outline-light-blue active:bg-light-blue-700",
    disabled: ""
  },
  {
    name: "cyan",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-cyan-600",
    active: "hover:bg-cyan-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-cyan-500 focus:shadow-outline-cyan active:bg-cyan-700",
    disabled: ""
  },
  {
    name: "purple",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-purple-600",
    active: "hover:bg-purple-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500 focus:shadow-outline-purple active:bg-purple-700",
    disabled: ""
  },
  {
    name: "pink",
    primary: "border border-transparent font-medium rounded shadow-sm text-white bg-pink-600",
    active: "hover:bg-pink-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-pink-500 focus:shadow-outline-pink active:bg-pink-700",
    disabled: ""
  }
], yt = {
  key: 0,
  class: "absolute top-0 right-0 -mr-1 -mt-1 z-10"
}, pt = { class: "flex w-3 h-3 relative" }, ye = /* @__PURE__ */ E({
  __name: "PushButton",
  props: {
    alwaysClick: Boolean,
    customTheme: Object,
    theme: String,
    state: {
      type: String,
      default: "active"
    },
    size: {
      type: String,
      default: "m"
    },
    ping: Boolean,
    pingColor: {
      type: String,
      default: "bg-indigo-500"
    }
  },
  emits: ["click"],
  setup(e, { emit: t }) {
    const n = e, s = F(() => n.state === "active"), o = F(() => n.state === "disabled"), r = F(() => s.value ? "cursor-pointer" : o.value ? "cursor-not-allowed" : "cursor-wait"), i = {
      xs: "px-2.5 py-1.5 text-xs leading-4",
      s: "px-3 py-2 text-sm leading-4",
      m: "px-4 py-2 text-sm leading-5",
      l: "px-4 py-2 text-base leading-6",
      xl: "px-6 py-3 text-base leading-6"
    }, a = F(() => n.customTheme ? n.customTheme : n.theme && se.find(({ name: c }) => c === n.theme) ? se.find(({ name: c }) => c === n.theme) : se.find(({ name: c }) => c === "white"));
    function l() {
      s && t("click");
    }
    return (c, u) => (h(), y("button", {
      type: "button",
      class: S(["relative inline-flex items-center", [i[e.size], p(a).primary, p(a).dark, p(r), p(s) ? p(a).active : p(a).disabled]]),
      onClick: l
    }, [
      Re(c.$slots, "default"),
      e.ping ? (h(), y("span", yt, [
        d("span", pt, [
          d("span", {
            class: S(["animate-ping absolute inline-flex h-full w-full rounded-full opacity-75", e.pingColor])
          }, null, 2),
          d("span", {
            class: S(["relative inline-flex rounded-full h-3 w-3", e.pingColor])
          }, null, 2)
        ])
      ])) : v("", !0)
    ], 2));
  }
});
function bt(e) {
  return ut() ? (ft(e), !0) : !1;
}
const je = typeof window < "u", vt = (e) => typeof e == "string", ie = () => {
}, Ue = je ? window : void 0;
je && window.document;
je && window.navigator;
function wt(...e) {
  let t, n, s, o;
  if (vt(e[0]) ? ([n, s, o] = e, t = Ue) : [t, n, s, o] = e, !t)
    return ie;
  let r = ie;
  const i = dt(() => p(t), (l) => {
    r(), l && (l.addEventListener(n, s, o), r = () => {
      l.removeEventListener(n, s, o), r = ie;
    });
  }, { immediate: !0, flush: "post" }), a = () => {
    i(), r();
  };
  return bt(a), a;
}
const xt = (e) => typeof e == "function" ? e : typeof e == "string" ? (t) => t.key === e : Array.isArray(e) ? (t) => e.includes(t.key) : e ? () => !0 : () => !1;
function kt(e, t, n = {}) {
  const { target: s = Ue, eventName: o = "keydown", passive: r = !1 } = n, i = xt(e);
  return wt(s, o, (l) => {
    i(l) && t(l);
  }, r);
}
var Me;
(function(e) {
  e.UP = "UP", e.RIGHT = "RIGHT", e.DOWN = "DOWN", e.LEFT = "LEFT", e.NONE = "NONE";
})(Me || (Me = {}));
const G = /^[a-z0-9]+(-[a-z0-9]+)*$/, T = Object.freeze({
  left: 0,
  top: 0,
  width: 16,
  height: 16,
  rotate: 0,
  vFlip: !1,
  hFlip: !1
});
function ne(e) {
  return { ...T, ...e };
}
const oe = (e, t, n, s = "") => {
  const o = e.split(":");
  if (e.slice(0, 1) === "@") {
    if (o.length < 2 || o.length > 3)
      return null;
    s = o.shift().slice(1);
  }
  if (o.length > 3 || !o.length)
    return null;
  if (o.length > 1) {
    const a = o.pop(), l = o.pop(), c = {
      provider: o.length > 0 ? o[0] : s,
      prefix: l,
      name: a
    };
    return t && !Q(c) ? null : c;
  }
  const r = o[0], i = r.split("-");
  if (i.length > 1) {
    const a = {
      provider: s,
      prefix: i.shift(),
      name: i.join("-")
    };
    return t && !Q(a) ? null : a;
  }
  if (n && s === "") {
    const a = {
      provider: s,
      prefix: "",
      name: r
    };
    return t && !Q(a, n) ? null : a;
  }
  return null;
}, Q = (e, t) => e ? !!((e.provider === "" || e.provider.match(G)) && (t && e.prefix === "" || e.prefix.match(G)) && e.name.match(G)) : !1;
function It(e, t) {
  const n = { ...e };
  for (const s in T) {
    const o = s;
    if (t[o] !== void 0) {
      const r = t[o];
      if (n[o] === void 0) {
        n[o] = r;
        continue;
      }
      switch (o) {
        case "rotate":
          n[o] = (n[o] + r) % 4;
          break;
        case "hFlip":
        case "vFlip":
          n[o] = r !== n[o];
          break;
        default:
          n[o] = r;
      }
    }
  }
  return n;
}
function Le(e, t, n = !1) {
  function s(r, i) {
    if (e.icons[r] !== void 0)
      return Object.assign({}, e.icons[r]);
    if (i > 5)
      return null;
    const a = e.aliases;
    if (a && a[r] !== void 0) {
      const c = a[r], u = s(c.parent, i + 1);
      return u && It(u, c);
    }
    const l = e.chars;
    return !i && l && l[r] !== void 0 ? s(l[r], i + 1) : null;
  }
  const o = s(t, 0);
  if (o)
    for (const r in T)
      o[r] === void 0 && e[r] !== void 0 && (o[r] = e[r]);
  return o && n ? ne(o) : o;
}
function jt(e) {
  for (const t in T)
    if (e[t] !== void 0)
      return !0;
  return !1;
}
function Ve(e, t, n) {
  n = n || {};
  const s = [];
  if (typeof e != "object" || typeof e.icons != "object")
    return s;
  e.not_found instanceof Array && e.not_found.forEach((i) => {
    t(i, null), s.push(i);
  });
  const o = e.icons;
  Object.keys(o).forEach((i) => {
    const a = Le(e, i, !0);
    a && (t(i, a), s.push(i));
  });
  const r = n.aliases || "all";
  if (r !== "none" && typeof e.aliases == "object") {
    const i = e.aliases;
    Object.keys(i).forEach((a) => {
      if (r === "variations" && jt(i[a]))
        return;
      const l = Le(e, a, !0);
      l && (t(a, l), s.push(a));
    });
  }
  return s;
}
const pe = {
  provider: "string",
  aliases: "object",
  not_found: "object"
};
for (const e in T)
  pe[e] = typeof T[e];
function Ge(e) {
  if (typeof e != "object" || e === null)
    return null;
  const t = e;
  if (typeof t.prefix != "string" || !e.icons || typeof e.icons != "object")
    return null;
  for (const o in pe)
    if (e[o] !== void 0 && typeof e[o] !== pe[o])
      return null;
  const n = t.icons;
  for (const o in n) {
    const r = n[o];
    if (!o.match(G) || typeof r.body != "string")
      return null;
    for (const i in T)
      if (r[i] !== void 0 && typeof r[i] != typeof T[i])
        return null;
  }
  const s = t.aliases;
  if (s)
    for (const o in s) {
      const r = s[o], i = r.parent;
      if (!o.match(G) || typeof i != "string" || !n[i] && !s[i])
        return null;
      for (const a in T)
        if (r[a] !== void 0 && typeof r[a] != typeof T[a])
          return null;
    }
  return t;
}
const Ct = 1;
let Z = /* @__PURE__ */ Object.create(null);
try {
  const e = window || self;
  e && e._iconifyStorage.version === Ct && (Z = e._iconifyStorage.storage);
} catch {
}
function St(e, t) {
  return {
    provider: e,
    prefix: t,
    icons: /* @__PURE__ */ Object.create(null),
    missing: /* @__PURE__ */ Object.create(null)
  };
}
function B(e, t) {
  Z[e] === void 0 && (Z[e] = /* @__PURE__ */ Object.create(null));
  const n = Z[e];
  return n[t] === void 0 && (n[t] = St(e, t)), n[t];
}
function Ce(e, t) {
  if (!Ge(t))
    return [];
  const n = Date.now();
  return Ve(t, (s, o) => {
    o ? e.icons[s] = o : e.missing[s] = n;
  });
}
function Tt(e, t, n) {
  try {
    if (typeof n.body == "string")
      return e.icons[t] = Object.freeze(ne(n)), !0;
  } catch {
  }
  return !1;
}
function Et(e, t) {
  const n = e.icons[t];
  return n === void 0 ? null : n;
}
let K = !1;
function Qe(e) {
  return typeof e == "boolean" && (K = e), K;
}
function Ot(e) {
  const t = typeof e == "string" ? oe(e, !0, K) : e;
  return t ? Et(B(t.provider, t.prefix), t.name) : null;
}
function _t(e, t) {
  const n = oe(e, !0, K);
  if (!n)
    return !1;
  const s = B(n.provider, n.prefix);
  return Tt(s, n.name, t);
}
function $t(e, t) {
  if (typeof e != "object")
    return !1;
  if (typeof t != "string" && (t = typeof e.provider == "string" ? e.provider : ""), K && t === "" && (typeof e.prefix != "string" || e.prefix === "")) {
    let s = !1;
    return Ge(e) && (e.prefix = "", Ve(e, (o, r) => {
      r && _t(o, r) && (s = !0);
    })), s;
  }
  if (typeof e.prefix != "string" || !Q({
    provider: t,
    prefix: e.prefix,
    name: "a"
  }))
    return !1;
  const n = B(t, e.prefix);
  return !!Ce(n, e);
}
const Pe = Object.freeze({
  inline: !1,
  width: null,
  height: null,
  hAlign: "center",
  vAlign: "middle",
  slice: !1,
  hFlip: !1,
  vFlip: !1,
  rotate: 0
});
function At(e, t) {
  const n = {};
  for (const s in e) {
    const o = s;
    if (n[o] = e[o], t[o] === void 0)
      continue;
    const r = t[o];
    switch (o) {
      case "inline":
      case "slice":
        typeof r == "boolean" && (n[o] = r);
        break;
      case "hFlip":
      case "vFlip":
        r === !0 && (n[o] = !n[o]);
        break;
      case "hAlign":
      case "vAlign":
        typeof r == "string" && r !== "" && (n[o] = r);
        break;
      case "width":
      case "height":
        (typeof r == "string" && r !== "" || typeof r == "number" && r || r === null) && (n[o] = r);
        break;
      case "rotate":
        typeof r == "number" && (n[o] += r);
        break;
    }
  }
  return n;
}
const Mt = /(-?[0-9.]*[0-9]+[0-9.]*)/g, Lt = /^-?[0-9.]*[0-9]+[0-9.]*$/g;
function ae(e, t, n) {
  if (t === 1)
    return e;
  if (n = n === void 0 ? 100 : n, typeof e == "number")
    return Math.ceil(e * t * n) / n;
  if (typeof e != "string")
    return e;
  const s = e.split(Mt);
  if (s === null || !s.length)
    return e;
  const o = [];
  let r = s.shift(), i = Lt.test(r);
  for (; ; ) {
    if (i) {
      const a = parseFloat(r);
      isNaN(a) ? o.push(r) : o.push(Math.ceil(a * t * n) / n);
    } else
      o.push(r);
    if (r = s.shift(), r === void 0)
      return o.join("");
    i = !i;
  }
}
function Pt(e) {
  let t = "";
  switch (e.hAlign) {
    case "left":
      t += "xMin";
      break;
    case "right":
      t += "xMax";
      break;
    default:
      t += "xMid";
  }
  switch (e.vAlign) {
    case "top":
      t += "YMin";
      break;
    case "bottom":
      t += "YMax";
      break;
    default:
      t += "YMid";
  }
  return t += e.slice ? " slice" : " meet", t;
}
function Nt(e, t) {
  const n = {
    left: e.left,
    top: e.top,
    width: e.width,
    height: e.height
  };
  let s = e.body;
  [e, t].forEach((a) => {
    const l = [], c = a.hFlip, u = a.vFlip;
    let f = a.rotate;
    c ? u ? f += 2 : (l.push("translate(" + (n.width + n.left).toString() + " " + (0 - n.top).toString() + ")"), l.push("scale(-1 1)"), n.top = n.left = 0) : u && (l.push("translate(" + (0 - n.left).toString() + " " + (n.height + n.top).toString() + ")"), l.push("scale(1 -1)"), n.top = n.left = 0);
    let g;
    switch (f < 0 && (f -= Math.floor(f / 4) * 4), f = f % 4, f) {
      case 1:
        g = n.height / 2 + n.top, l.unshift("rotate(90 " + g.toString() + " " + g.toString() + ")");
        break;
      case 2:
        l.unshift("rotate(180 " + (n.width / 2 + n.left).toString() + " " + (n.height / 2 + n.top).toString() + ")");
        break;
      case 3:
        g = n.width / 2 + n.left, l.unshift("rotate(-90 " + g.toString() + " " + g.toString() + ")");
        break;
    }
    f % 2 === 1 && ((n.left !== 0 || n.top !== 0) && (g = n.left, n.left = n.top, n.top = g), n.width !== n.height && (g = n.width, n.width = n.height, n.height = g)), l.length && (s = '<g transform="' + l.join(" ") + '">' + s + "</g>");
  });
  let o, r;
  t.width === null && t.height === null ? (r = "1em", o = ae(r, n.width / n.height)) : t.width !== null && t.height !== null ? (o = t.width, r = t.height) : t.height !== null ? (r = t.height, o = ae(r, n.width / n.height)) : (o = t.width, r = ae(o, n.height / n.width)), o === "auto" && (o = n.width), r === "auto" && (r = n.height), o = typeof o == "string" ? o : o.toString() + "", r = typeof r == "string" ? r : r.toString() + "";
  const i = {
    attributes: {
      width: o,
      height: r,
      preserveAspectRatio: Pt(t),
      viewBox: n.left.toString() + " " + n.top.toString() + " " + n.width.toString() + " " + n.height.toString()
    },
    body: s
  };
  return t.inline && (i.inline = !0), i;
}
const Ft = /\sid="(\S+)"/g, Dt = "IconifyId" + Date.now().toString(16) + (Math.random() * 16777216 | 0).toString(16);
let Bt = 0;
function qt(e, t = Dt) {
  const n = [];
  let s;
  for (; s = Ft.exec(e); )
    n.push(s[1]);
  return n.length && n.forEach((o) => {
    const r = typeof t == "function" ? t(o) : t + (Bt++).toString(), i = o.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
    e = e.replace(new RegExp('([#;"])(' + i + ')([")]|\\.[a-z])', "g"), "$1" + r + "$3");
  }), e;
}
const be = /* @__PURE__ */ Object.create(null);
function zt(e, t) {
  be[e] = t;
}
function ve(e) {
  return be[e] || be[""];
}
function Se(e) {
  let t;
  if (typeof e.resources == "string")
    t = [e.resources];
  else if (t = e.resources, !(t instanceof Array) || !t.length)
    return null;
  return {
    resources: t,
    path: e.path === void 0 ? "/" : e.path,
    maxURL: e.maxURL ? e.maxURL : 500,
    rotate: e.rotate ? e.rotate : 750,
    timeout: e.timeout ? e.timeout : 5e3,
    random: e.random === !0,
    index: e.index ? e.index : 0,
    dataAfterTimeout: e.dataAfterTimeout !== !1
  };
}
const Te = /* @__PURE__ */ Object.create(null), V = [
  "https://api.simplesvg.com",
  "https://api.unisvg.com"
], ee = [];
for (; V.length > 0; )
  V.length === 1 || Math.random() > 0.5 ? ee.push(V.shift()) : ee.push(V.pop());
Te[""] = Se({
  resources: ["https://api.iconify.design"].concat(ee)
});
function Ht(e, t) {
  const n = Se(t);
  return n === null ? !1 : (Te[e] = n, !0);
}
function Ee(e) {
  return Te[e];
}
const Ke = (e, t) => {
  let n = e, s = n.indexOf("?") !== -1;
  function o(r) {
    switch (typeof r) {
      case "boolean":
        return r ? "true" : "false";
      case "number":
        return encodeURIComponent(r);
      case "string":
        return encodeURIComponent(r);
      default:
        throw new Error("Invalid parameter");
    }
  }
  return Object.keys(t).forEach((r) => {
    let i;
    try {
      i = o(t[r]);
    } catch {
      return;
    }
    n += (s ? "&" : "?") + encodeURIComponent(r) + "=" + i, s = !0;
  }), n;
}, We = {}, te = {}, Rt = () => {
  let e;
  try {
    if (e = fetch, typeof e == "function")
      return e;
  } catch {
  }
  return null;
};
let Ne = Rt();
function Ut(e, t) {
  const n = Ee(e);
  if (!n)
    return 0;
  let s;
  if (!n.maxURL)
    s = 0;
  else {
    let r = 0;
    n.resources.forEach((a) => {
      r = Math.max(r, a.length);
    });
    const i = Ke(t + ".json", {
      icons: ""
    });
    s = n.maxURL - r - n.path.length - i.length;
  }
  const o = e + ":" + t;
  return te[e] = n.path, We[o] = s, s;
}
function Vt(e) {
  return e === 404;
}
const Gt = (e, t, n) => {
  const s = [];
  let o = We[t];
  o === void 0 && (o = Ut(e, t));
  const r = "icons";
  let i = {
    type: r,
    provider: e,
    prefix: t,
    icons: []
  }, a = 0;
  return n.forEach((l, c) => {
    a += l.length + 1, a >= o && c > 0 && (s.push(i), i = {
      type: r,
      provider: e,
      prefix: t,
      icons: []
    }, a = l.length), i.icons.push(l);
  }), s.push(i), s;
};
function Qt(e) {
  if (typeof e == "string") {
    if (te[e] === void 0) {
      const t = Ee(e);
      if (!t)
        return "/";
      te[e] = t.path;
    }
    return te[e];
  }
  return "/";
}
const Kt = (e, t, n) => {
  if (!Ne) {
    n("abort", 424);
    return;
  }
  let s = Qt(t.provider);
  switch (t.type) {
    case "icons": {
      const r = t.prefix, a = t.icons.join(",");
      s += Ke(r + ".json", {
        icons: a
      });
      break;
    }
    case "custom": {
      const r = t.uri;
      s += r.slice(0, 1) === "/" ? r.slice(1) : r;
      break;
    }
    default:
      n("abort", 400);
      return;
  }
  let o = 503;
  Ne(e + s).then((r) => {
    const i = r.status;
    if (i !== 200) {
      setTimeout(() => {
        n(Vt(i) ? "abort" : "next", i);
      });
      return;
    }
    return o = 501, r.json();
  }).then((r) => {
    if (typeof r != "object" || r === null) {
      setTimeout(() => {
        n("next", o);
      });
      return;
    }
    setTimeout(() => {
      n("success", r);
    });
  }).catch(() => {
    n("next", o);
  });
}, Wt = {
  prepare: Gt,
  send: Kt
};
function Yt(e) {
  const t = {
    loaded: [],
    missing: [],
    pending: []
  }, n = /* @__PURE__ */ Object.create(null);
  e.sort((o, r) => o.provider !== r.provider ? o.provider.localeCompare(r.provider) : o.prefix !== r.prefix ? o.prefix.localeCompare(r.prefix) : o.name.localeCompare(r.name));
  let s = {
    provider: "",
    prefix: "",
    name: ""
  };
  return e.forEach((o) => {
    if (s.name === o.name && s.prefix === o.prefix && s.provider === o.provider)
      return;
    s = o;
    const r = o.provider, i = o.prefix, a = o.name;
    n[r] === void 0 && (n[r] = /* @__PURE__ */ Object.create(null));
    const l = n[r];
    l[i] === void 0 && (l[i] = B(r, i));
    const c = l[i];
    let u;
    c.icons[a] !== void 0 ? u = t.loaded : i === "" || c.missing[a] !== void 0 ? u = t.missing : u = t.pending;
    const f = {
      provider: r,
      prefix: i,
      name: a
    };
    u.push(f);
  }), t;
}
const M = /* @__PURE__ */ Object.create(null), le = /* @__PURE__ */ Object.create(null);
function Ye(e, t) {
  e.forEach((n) => {
    const s = n.provider;
    if (M[s] === void 0)
      return;
    const o = M[s], r = n.prefix, i = o[r];
    i && (o[r] = i.filter((a) => a.id !== t));
  });
}
function Jt(e, t) {
  le[e] === void 0 && (le[e] = /* @__PURE__ */ Object.create(null));
  const n = le[e];
  n[t] || (n[t] = !0, setTimeout(() => {
    if (n[t] = !1, M[e] === void 0 || M[e][t] === void 0)
      return;
    const s = M[e][t].slice(0);
    if (!s.length)
      return;
    const o = B(e, t);
    let r = !1;
    s.forEach((i) => {
      const a = i.icons, l = a.pending.length;
      a.pending = a.pending.filter((c) => {
        if (c.prefix !== t)
          return !0;
        const u = c.name;
        if (o.icons[u] !== void 0)
          a.loaded.push({
            provider: e,
            prefix: t,
            name: u
          });
        else if (o.missing[u] !== void 0)
          a.missing.push({
            provider: e,
            prefix: t,
            name: u
          });
        else
          return r = !0, !0;
        return !1;
      }), a.pending.length !== l && (r || Ye([
        {
          provider: e,
          prefix: t
        }
      ], i.id), i.callback(a.loaded.slice(0), a.missing.slice(0), a.pending.slice(0), i.abort));
    });
  }));
}
let Xt = 0;
function Zt(e, t, n) {
  const s = Xt++, o = Ye.bind(null, n, s);
  if (!t.pending.length)
    return o;
  const r = {
    id: s,
    icons: t,
    callback: e,
    abort: o
  };
  return n.forEach((i) => {
    const a = i.provider, l = i.prefix;
    M[a] === void 0 && (M[a] = /* @__PURE__ */ Object.create(null));
    const c = M[a];
    c[l] === void 0 && (c[l] = []), c[l].push(r);
  }), o;
}
function en(e, t = !0, n = !1) {
  const s = [];
  return e.forEach((o) => {
    const r = typeof o == "string" ? oe(o, !1, n) : o;
    (!t || Q(r, n)) && s.push({
      provider: r.provider,
      prefix: r.prefix,
      name: r.name
    });
  }), s;
}
var Fe = {
  resources: [],
  index: 0,
  timeout: 2e3,
  rotate: 750,
  random: !1,
  dataAfterTimeout: !1
};
function tn(e, t, n, s) {
  const o = e.resources.length, r = e.random ? Math.floor(Math.random() * o) : e.index;
  let i;
  if (e.random) {
    let m = e.resources.slice(0);
    for (i = []; m.length > 1; ) {
      const k = Math.floor(Math.random() * m.length);
      i.push(m[k]), m = m.slice(0, k).concat(m.slice(k + 1));
    }
    i = i.concat(m);
  } else
    i = e.resources.slice(r).concat(e.resources.slice(0, r));
  const a = Date.now();
  let l = "pending", c = 0, u, f = null, g = [], b = [];
  typeof s == "function" && b.push(s);
  function x() {
    f && (clearTimeout(f), f = null);
  }
  function L() {
    l === "pending" && (l = "aborted"), x(), g.forEach((m) => {
      m.status === "pending" && (m.status = "aborted");
    }), g = [];
  }
  function q(m, k) {
    k && (b = []), typeof m == "function" && b.push(m);
  }
  function j() {
    return {
      startTime: a,
      payload: t,
      status: l,
      queriesSent: c,
      queriesPending: g.length,
      subscribe: q,
      abort: L
    };
  }
  function O() {
    l = "failed", b.forEach((m) => {
      m(void 0, u);
    });
  }
  function $e() {
    g.forEach((m) => {
      m.status === "pending" && (m.status = "aborted");
    }), g = [];
  }
  function ct(m, k, U) {
    const Y = k !== "success";
    switch (g = g.filter((P) => P !== m), l) {
      case "pending":
        break;
      case "failed":
        if (Y || !e.dataAfterTimeout)
          return;
        break;
      default:
        return;
    }
    if (k === "abort") {
      u = U, O();
      return;
    }
    if (Y) {
      u = U, g.length || (i.length ? re() : O());
      return;
    }
    if (x(), $e(), !e.random) {
      const P = e.resources.indexOf(m.resource);
      P !== -1 && P !== e.index && (e.index = P);
    }
    l = "completed", b.forEach((P) => {
      P(U);
    });
  }
  function re() {
    if (l !== "pending")
      return;
    x();
    const m = i.shift();
    if (m === void 0) {
      if (g.length) {
        f = setTimeout(() => {
          x(), l === "pending" && ($e(), O());
        }, e.timeout);
        return;
      }
      O();
      return;
    }
    const k = {
      status: "pending",
      resource: m,
      callback: (U, Y) => {
        ct(k, U, Y);
      }
    };
    g.push(k), c++, f = setTimeout(re, e.rotate), n(m, t, k.callback);
  }
  return setTimeout(re), j;
}
function nn(e) {
  if (typeof e != "object" || typeof e.resources != "object" || !(e.resources instanceof Array) || !e.resources.length)
    throw new Error("Invalid Reduncancy configuration");
  const t = /* @__PURE__ */ Object.create(null);
  let n;
  for (n in Fe)
    e[n] !== void 0 ? t[n] = e[n] : t[n] = Fe[n];
  return t;
}
function Je(e) {
  const t = nn(e);
  let n = [];
  function s() {
    n = n.filter((a) => a().status === "pending");
  }
  function o(a, l, c) {
    const u = tn(t, a, l, (f, g) => {
      s(), c && c(f, g);
    });
    return n.push(u), u;
  }
  function r(a) {
    const l = n.find((c) => a(c));
    return l !== void 0 ? l : null;
  }
  return {
    query: o,
    find: r,
    setIndex: (a) => {
      t.index = a;
    },
    getIndex: () => t.index,
    cleanup: s
  };
}
function De() {
}
const ce = /* @__PURE__ */ Object.create(null);
function on(e) {
  if (ce[e] === void 0) {
    const t = Ee(e);
    if (!t)
      return;
    const n = Je(t), s = {
      config: t,
      redundancy: n
    };
    ce[e] = s;
  }
  return ce[e];
}
function rn(e, t, n) {
  let s, o;
  if (typeof e == "string") {
    const r = ve(e);
    if (!r)
      return n(void 0, 424), De;
    o = r.send;
    const i = on(e);
    i && (s = i.redundancy);
  } else {
    const r = Se(e);
    if (r) {
      s = Je(r);
      const i = e.resources ? e.resources[0] : "", a = ve(i);
      a && (o = a.send);
    }
  }
  return !s || !o ? (n(void 0, 424), De) : s.query(t, o, n)().abort;
}
const we = {};
function Be() {
}
const D = /* @__PURE__ */ Object.create(null), ue = /* @__PURE__ */ Object.create(null), fe = /* @__PURE__ */ Object.create(null), de = /* @__PURE__ */ Object.create(null);
function sn(e, t) {
  fe[e] === void 0 && (fe[e] = /* @__PURE__ */ Object.create(null));
  const n = fe[e];
  n[t] || (n[t] = !0, setTimeout(() => {
    n[t] = !1, Jt(e, t);
  }));
}
const qe = /* @__PURE__ */ Object.create(null);
function an(e, t, n) {
  function s() {
    const a = (e === "" ? "" : "@" + e + ":") + t, l = Math.floor(Date.now() / 6e4);
    qe[a] < l && (qe[a] = l, console.error('Unable to retrieve icons for "' + a + '" because API is not configured properly.'));
  }
  ue[e] === void 0 && (ue[e] = /* @__PURE__ */ Object.create(null));
  const o = ue[e];
  de[e] === void 0 && (de[e] = /* @__PURE__ */ Object.create(null));
  const r = de[e];
  D[e] === void 0 && (D[e] = /* @__PURE__ */ Object.create(null));
  const i = D[e];
  o[t] === void 0 ? o[t] = n : o[t] = o[t].concat(n).sort(), r[t] || (r[t] = !0, setTimeout(() => {
    r[t] = !1;
    const a = o[t];
    delete o[t];
    const l = ve(e);
    if (!l) {
      s();
      return;
    }
    l.prepare(e, t, a).forEach((u) => {
      rn(e, u, (f, g) => {
        const b = B(e, t);
        if (typeof f != "object") {
          if (g !== 404)
            return;
          const x = Date.now();
          u.icons.forEach((L) => {
            b.missing[L] = x;
          });
        } else
          try {
            const x = Ce(b, f);
            if (!x.length)
              return;
            const L = i[t];
            x.forEach((q) => {
              delete L[q];
            }), we.store && we.store(e, f);
          } catch (x) {
            console.error(x);
          }
        sn(e, t);
      });
    });
  }));
}
const ln = (e, t) => {
  const n = en(e, !0, Qe()), s = Yt(n);
  if (!s.pending.length) {
    let c = !0;
    return t && setTimeout(() => {
      c && t(s.loaded, s.missing, s.pending, Be);
    }), () => {
      c = !1;
    };
  }
  const o = /* @__PURE__ */ Object.create(null), r = [];
  let i, a;
  s.pending.forEach((c) => {
    const u = c.provider, f = c.prefix;
    if (f === a && u === i)
      return;
    i = u, a = f, r.push({
      provider: u,
      prefix: f
    }), D[u] === void 0 && (D[u] = /* @__PURE__ */ Object.create(null));
    const g = D[u];
    g[f] === void 0 && (g[f] = /* @__PURE__ */ Object.create(null)), o[u] === void 0 && (o[u] = /* @__PURE__ */ Object.create(null));
    const b = o[u];
    b[f] === void 0 && (b[f] = []);
  });
  const l = Date.now();
  return s.pending.forEach((c) => {
    const u = c.provider, f = c.prefix, g = c.name, b = D[u][f];
    b[g] === void 0 && (b[g] = l, o[u][f].push(g));
  }), r.forEach((c) => {
    const u = c.provider, f = c.prefix;
    o[u][f].length && an(u, f, o[u][f]);
  }), t ? Zt(t, s, r) : Be;
}, Xe = "iconify2", W = "iconify", Ze = W + "-count", et = W + "-version", tt = 36e5, cn = 168, Oe = {
  local: !0,
  session: !0
};
let xe = !1;
const nt = {
  local: 0,
  session: 0
}, ot = {
  local: [],
  session: []
};
let J = typeof window > "u" ? {} : window;
function rt(e) {
  const t = e + "Storage";
  try {
    if (J && J[t] && typeof J[t].length == "number")
      return J[t];
  } catch {
  }
  return Oe[e] = !1, null;
}
function _e(e, t, n) {
  try {
    return e.setItem(Ze, n.toString()), nt[t] = n, !0;
  } catch {
    return !1;
  }
}
function st(e) {
  const t = e.getItem(Ze);
  if (t) {
    const n = parseInt(t);
    return n || 0;
  }
  return 0;
}
function un(e, t) {
  try {
    e.setItem(et, Xe);
  } catch {
  }
  _e(e, t, 0);
}
function fn(e) {
  try {
    const t = st(e);
    for (let n = 0; n < t; n++)
      e.removeItem(W + n.toString());
  } catch {
  }
}
const it = () => {
  if (xe)
    return;
  xe = !0;
  const e = Math.floor(Date.now() / tt) - cn;
  function t(n) {
    const s = rt(n);
    if (!s)
      return;
    const o = (r) => {
      const i = W + r.toString(), a = s.getItem(i);
      if (typeof a != "string")
        return !1;
      let l = !0;
      try {
        const c = JSON.parse(a);
        if (typeof c != "object" || typeof c.cached != "number" || c.cached < e || typeof c.provider != "string" || typeof c.data != "object" || typeof c.data.prefix != "string")
          l = !1;
        else {
          const u = c.provider, f = c.data.prefix, g = B(u, f);
          l = Ce(g, c.data).length > 0;
        }
      } catch {
        l = !1;
      }
      return l || s.removeItem(i), l;
    };
    try {
      const r = s.getItem(et);
      if (r !== Xe) {
        r && fn(s), un(s, n);
        return;
      }
      let i = st(s);
      for (let a = i - 1; a >= 0; a--)
        o(a) || (a === i - 1 ? i-- : ot[n].push(a));
      _e(s, n, i);
    } catch {
    }
  }
  for (const n in Oe)
    t(n);
}, dn = (e, t) => {
  xe || it();
  function n(s) {
    if (!Oe[s])
      return !1;
    const o = rt(s);
    if (!o)
      return !1;
    let r = ot[s].shift();
    if (r === void 0 && (r = nt[s], !_e(o, s, r + 1)))
      return !1;
    try {
      const i = {
        cached: Math.floor(Date.now() / tt),
        provider: e,
        data: t
      };
      o.setItem(W + r.toString(), JSON.stringify(i));
    } catch {
      return !1;
    }
    return !0;
  }
  !Object.keys(t.icons).length || (t.not_found && (t = Object.assign({}, t), delete t.not_found), n("local") || n("session"));
}, at = /[\s,]+/;
function gn(e, t) {
  t.split(at).forEach((n) => {
    switch (n.trim()) {
      case "horizontal":
        e.hFlip = !0;
        break;
      case "vertical":
        e.vFlip = !0;
        break;
    }
  });
}
function hn(e, t) {
  t.split(at).forEach((n) => {
    const s = n.trim();
    switch (s) {
      case "left":
      case "center":
      case "right":
        e.hAlign = s;
        break;
      case "top":
      case "middle":
      case "bottom":
        e.vAlign = s;
        break;
      case "slice":
      case "crop":
        e.slice = !0;
        break;
      case "meet":
        e.slice = !1;
    }
  });
}
function mn(e, t = 0) {
  const n = e.replace(/^-?[0-9.]*/, "");
  function s(o) {
    for (; o < 0; )
      o += 4;
    return o % 4;
  }
  if (n === "") {
    const o = parseInt(e);
    return isNaN(o) ? 0 : s(o);
  } else if (n !== e) {
    let o = 0;
    switch (n) {
      case "%":
        o = 25;
        break;
      case "deg":
        o = 90;
    }
    if (o) {
      let r = parseFloat(e.slice(0, e.length - n.length));
      return isNaN(r) ? 0 : (r = r / o, r % 1 === 0 ? s(r) : 0);
    }
  }
  return t;
}
const yn = {
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  "aria-hidden": !0,
  role: "img"
};
let $ = {};
["horizontal", "vertical"].forEach((e) => {
  ["Align", "Flip"].forEach((t) => {
    const s = {
      attr: e.slice(0, 1) + t,
      boolean: t === "Flip"
    };
    $[e + "-" + t.toLowerCase()] = s, $[e.slice(0, 1) + "-" + t.toLowerCase()] = s, $[e + t] = s;
  });
});
const ze = (e, t) => {
  const n = At(Pe, t), s = { ...yn };
  let o = typeof t.style == "object" && !(t.style instanceof Array) ? { ...t.style } : {};
  for (let l in t) {
    const c = t[l];
    if (c !== void 0)
      switch (l) {
        case "icon":
        case "style":
        case "onLoad":
          break;
        case "inline":
        case "hFlip":
        case "vFlip":
          n[l] = c === !0 || c === "true" || c === 1;
          break;
        case "flip":
          typeof c == "string" && gn(n, c);
          break;
        case "align":
          typeof c == "string" && hn(n, c);
          break;
        case "color":
          o.color = c;
          break;
        case "rotate":
          typeof c == "string" ? n[l] = mn(c) : typeof c == "number" && (n[l] = c);
          break;
        case "ariaHidden":
        case "aria-hidden":
          c !== !0 && c !== "true" && delete s["aria-hidden"];
          break;
        default:
          $[l] !== void 0 ? $[l].boolean && (c === !0 || c === "true" || c === 1) ? n[$[l].attr] = !0 : !$[l].boolean && typeof c == "string" && c !== "" && (n[$[l].attr] = c) : Pe[l] === void 0 && (s[l] = c);
      }
  }
  const r = Nt(e, n);
  for (let l in r.attributes)
    s[l] = r.attributes[l];
  r.inline && o.verticalAlign === void 0 && o["vertical-align"] === void 0 && (o.verticalAlign = "-0.125em");
  let i = 0, a = t.id;
  return typeof a == "string" && (a = a.replace(/-/g, "_")), s.innerHTML = qt(r.body, a ? () => a + "ID" + i++ : "iconifyVue"), Object.keys(o).length > 0 && (s.style = o), gt("svg", s);
};
Qe(!0);
zt("", Wt);
if (typeof document < "u" && typeof window < "u") {
  we.store = dn, it();
  const e = window;
  if (e.IconifyPreload !== void 0) {
    const t = e.IconifyPreload, n = "Invalid IconifyPreload syntax.";
    typeof t == "object" && t !== null && (t instanceof Array ? t : [t]).forEach((s) => {
      try {
        (typeof s != "object" || s === null || s instanceof Array || typeof s.icons != "object" || typeof s.prefix != "string" || !$t(s)) && console.error(n);
      } catch {
        console.error(n);
      }
    });
  }
  if (e.IconifyProviders !== void 0) {
    const t = e.IconifyProviders;
    if (typeof t == "object" && t !== null)
      for (let n in t) {
        const s = "IconifyProviders[" + n + "] is invalid.";
        try {
          const o = t[n];
          if (typeof o != "object" || !o || o.resources === void 0)
            continue;
          Ht(n, o) || console.error(s);
        } catch {
          console.error(s);
        }
      }
  }
}
const pn = ne({
  body: ""
}), I = E({
  inheritAttrs: !1,
  data() {
    return {
      iconMounted: !1,
      counter: 0
    };
  },
  mounted() {
    this._name = "", this._loadingIcon = null, this.iconMounted = !0;
  },
  unmounted() {
    this.abortLoading();
  },
  methods: {
    abortLoading() {
      this._loadingIcon && (this._loadingIcon.abort(), this._loadingIcon = null);
    },
    getIcon(e, t) {
      if (typeof e == "object" && e !== null && typeof e.body == "string")
        return this._name = "", this.abortLoading(), {
          data: ne(e)
        };
      let n;
      if (typeof e != "string" || (n = oe(e, !1, !0)) === null)
        return this.abortLoading(), null;
      const s = Ot(n);
      if (s === null)
        return (!this._loadingIcon || this._loadingIcon.name !== e) && (this.abortLoading(), this._name = "", this._loadingIcon = {
          name: e,
          abort: ln([n], () => {
            this.counter++;
          })
        }), null;
      this.abortLoading(), this._name !== e && (this._name = e, t && t(e));
      const o = ["iconify"];
      return n.prefix !== "" && o.push("iconify--" + n.prefix), n.provider !== "" && o.push("iconify--" + n.provider), { data: s, classes: o };
    }
  },
  render() {
    this.counter;
    const e = this.$attrs, t = this.iconMounted ? this.getIcon(e.icon, e.onLoad) : null;
    if (!t)
      return ze(pn, e);
    let n = e;
    return t.classes && (n = {
      ...e,
      class: (typeof e.class == "string" ? e.class + " " : "") + t.classes.join(" ")
    }), ze(t.data, n);
  }
}), bn = {
  class: "fixed z-10 inset-0 overflow-y-auto",
  id: "ModalBase"
}, vn = { class: "flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0" }, wn = /* @__PURE__ */ d("div", { class: "absolute inset-0 bg-black opacity-75" }, null, -1), xn = { class: "fixed top-0 right-0 p-3" }, kn = { class: "p-3 cursor-pointer rounded-full transition ease-in-out duration-150 hover:bg-gray-800" }, In = /* @__PURE__ */ d("span", { class: "hidden sm:inline-block sm:align-middle sm:h-screen" }, "\xA0", -1), lt = /* @__PURE__ */ E({
  __name: "ModalBase",
  props: {
    destroyed: {
      type: [Function, () => {
      }],
      required: !1,
      default: () => {
      }
    },
    innerClass: {
      type: String,
      required: !1,
      default: ""
    },
    maxWidth: {
      type: String,
      required: !1,
      default: "sm:max-w-lg"
    }
  },
  setup(e, { expose: t }) {
    const n = e, s = A(!1);
    Ie(() => s.value = !0), kt("Escape", (r) => {
      r.preventDefault(), o();
    });
    async function o() {
      s.value = !1, setTimeout(() => n.destroyed(), 200);
    }
    return t({ destroy: o }), (r, i) => (h(), y("div", bn, [
      d("div", vn, [
        w(me, {
          "enter-active-class": "ease-out duration-300",
          "enter-from-class": "opacity-0",
          "enter-to-class": "opacity-100",
          "leave-active-class": "ease-in duration-200",
          "leave-class": "opacity-100",
          "leave-to-class": "opacity-0"
        }, {
          default: R(() => [
            s.value ? (h(), y("div", {
              key: 0,
              class: "fixed inset-0 transition-opacity",
              onClick: o
            }, [
              wn,
              d("div", xn, [
                d("div", kn, [
                  w(p(I), {
                    icon: "mdi-close",
                    class: "w-6 h-6 text-gray-200 dark:text-gray-400"
                  })
                ])
              ])
            ])) : v("", !0)
          ]),
          _: 1
        }),
        In,
        w(me, {
          "enter-active-class": "ease-out duration-300",
          "enter-from-class": "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95",
          "enter-to-class": "opacity-100 translate-y-0 sm:scale-100",
          "leave-active-class": "ease-in duration-200",
          "leave-class": "opacity-100 translate-y-0 sm:scale-100",
          "leave-to-class": "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
        }, {
          default: R(() => [
            s.value ? (h(), y("div", {
              key: 0,
              class: S(["inline-block align-bottom bg-white dark:bg-gray-800 rounded-lg px-4 pt-5 pb-4 text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:w-full sm:p-6", [e.innerClass, e.maxWidth]]),
              role: "dialog",
              "aria-modal": "true",
              "aria-labelledby": "modal-headline"
            }, [
              Re(r.$slots, "default")
            ], 2)) : v("", !0)
          ]),
          _: 3
        })
      ])
    ]));
  }
});
function ke(e) {
  !e || e.parentNode.remove();
}
function N(e, t, n, s) {
  var r;
  const o = document.createElement("div");
  return s && s.forEach((i) => o.classList.add(i)), ht(E(n), t).mount(
    (r = document == null ? void 0 : document.getElementById(e)) == null ? void 0 : r.appendChild(o)
  );
}
const jn = { class: "sm:flex sm:items-center" }, Cn = { class: "mt-3 text-center sm:mt-0 sm:ml-4 sm:text-left" }, Sn = {
  key: 0,
  class: "text-lg leading-6 font-medium text-gray-900 dark:text-gray-300 mb-1",
  id: "modal-headline"
}, Tn = ["innerHTML"], En = {
  key: 0,
  class: "mt-5 sm:mt-4 sm:flex sm:flex-row-reverse"
}, On = /* @__PURE__ */ E({
  __name: "GlobalModal",
  props: {
    type: {
      type: String,
      required: !1,
      default: "info"
    },
    title: {
      type: [Boolean, String],
      required: !1,
      default: !1
    },
    body: {
      type: [Boolean, String],
      required: !1,
      default: !1
    },
    primary: [Boolean, Object],
    secondary: [Boolean, Object]
  },
  setup(e) {
    const t = e, n = A(!1), s = A(null), o = A(null), r = {
      success: "bg-green-100 dark:bg-green-900",
      info: "bg-blue-100 dark:bg-blue-900",
      danger: "bg-red-100 dark:bg-red-900",
      warning: "bg-yellow-100 dark:bg-yellow-900"
    };
    Ie(() => {
      n.value = !0, setTimeout(() => {
        var l;
        t.primary && s && s.value && ((l = s == null ? void 0 : s.value) == null || l.$el.focus());
      }, 200);
    });
    async function i(l) {
      var c;
      o && ((c = o == null ? void 0 : o.value) == null || c.destroy()), l === "primary" && t.primary && t.primary.action(), l === "secondary" && t.secondary && t.secondary.action();
    }
    async function a() {
      var l;
      n.value = !1, ke((l = o == null ? void 0 : o.value) == null ? void 0 : l.$el);
    }
    return (l, c) => (h(), C(lt, {
      ref_key: "modalBaseRef",
      ref: o,
      active: n.value,
      destroyed: a
    }, {
      default: R(() => [
        d("div", jn, [
          d("div", {
            class: S([r[e.type], "mx-auto flex-shrink-0 self-start flex items-center justify-center h-12 w-12 rounded-full sm:mx-0 sm:h-10 sm:w-10"])
          }, [
            e.type === "success" ? (h(), C(p(I), {
              key: 0,
              icon: "mdi-check",
              class: "h-5 w-5 text-green-500"
            })) : v("", !0),
            e.type === "info" ? (h(), C(p(I), {
              key: 1,
              icon: "bi-info-lg",
              class: "h-5 w-5 text-blue-500"
            })) : v("", !0),
            e.type === "danger" ? (h(), C(p(I), {
              key: 2,
              icon: "mdi-exclamation-thick",
              class: "h-5 w-5 text-red-500"
            })) : v("", !0),
            e.type === "warning" ? (h(), C(p(I), {
              key: 3,
              icon: "mdi-exclamation-thick",
              class: "h-5 w-5 text-yellow-500"
            })) : v("", !0)
          ], 2),
          d("div", Cn, [
            e.title ? (h(), y("h3", Sn, _(e.title), 1)) : v("", !0),
            e.body ? (h(), y("p", {
              key: 1,
              class: "text-sm leading-5",
              innerHTML: e.body
            }, null, 8, Tn)) : v("", !0)
          ])
        ]),
        e.primary ? (h(), y("div", En, [
          e.primary ? (h(), C(ye, {
            key: 0,
            ref_key: "primaryRef",
            ref: s,
            class: "flex w-full rounded-md shadow-sm sm:ml-3 sm:w-auto",
            theme: e.primary.theme,
            onClick: c[0] || (c[0] = (u) => i("primary"))
          }, {
            default: R(() => [
              Ae(_(e.primary.label), 1)
            ]),
            _: 1
          }, 8, ["theme"])) : v("", !0),
          e.secondary ? (h(), C(ye, {
            key: 1,
            class: "mt-3 flex w-full rounded-md shadow-sm sm:mt-0 sm:w-auto",
            theme: e.secondary.theme,
            onClick: c[1] || (c[1] = (u) => i("secondary"))
          }, {
            default: R(() => [
              Ae(_(e.secondary.label), 1)
            ]),
            _: 1
          }, 8, ["theme"])) : v("", !0)
        ])) : v("", !0)
      ]),
      _: 1
    }, 8, ["active"]));
  }
});
function go() {
  const e = document.createElement("div");
  return e.classList.add("z-30"), e.setAttribute("id", "modals"), document.body.appendChild(e), {
    show(t) {
      return typeof t == "string" && (t = { title: t }), N("modals", t, On);
    }
  };
}
const _n = {
  key: 0,
  class: "border-2 border-green-200 rounded-full w-6 h-6 flex items-center justify-center"
}, $n = {
  key: 1,
  class: "border-2 border-blue-200 rounded-full w-6 h-6 flex items-center justify-center"
}, An = {
  key: 2,
  class: "border-2 border-yellow-200 rounded-full w-6 h-6 flex items-center justify-center"
}, Mn = {
  key: 3,
  class: "border-2 border-red-200 rounded-full w-6 h-6 flex items-center justify-center"
}, Ln = {
  key: 4,
  class: "rounded-full w-6 h-6 flex items-center justify-center"
}, Pn = {
  key: 5,
  class: "rounded-full w-6 h-6 flex items-center justify-center"
}, ge = /* @__PURE__ */ E({
  __name: "GlobalToastIcons",
  props: {
    type: String
  },
  setup(e) {
    const t = e;
    return (n, s) => t.type === "success" ? (h(), y("div", _n, [
      w(p(I), {
        icon: "mdi-check",
        class: "h-4 w-4 text-green-400"
      })
    ])) : t.type === "info" ? (h(), y("div", $n, [
      w(p(I), {
        icon: "bi-info-lg",
        class: "h-4 w-4 text-blue-400"
      })
    ])) : t.type === "warning" ? (h(), y("div", An, [
      w(p(I), {
        icon: "mdi-exclamation-thick",
        class: "h-4 w-4 text-yellow-400"
      })
    ])) : t.type === "danger" ? (h(), y("div", Mn, [
      w(p(I), {
        icon: "mdi-exclamation-thick",
        class: "h-4 w-4 text-red-400"
      })
    ])) : t.type === "denied" ? (h(), y("div", Ln, [
      w(p(I), {
        icon: "mdi-cancel",
        class: "h-4 w-4 text-red-400"
      })
    ])) : (h(), y("div", Pn, [
      w(p(I), {
        icon: e.type,
        class: "h-4 w-4 text-red-400"
      }, null, 8, ["icon"])
    ]));
  }
}), Nn = { class: "ml-4 flex-shrink-0 flex" }, Fn = /* @__PURE__ */ d("span", { class: "sr-only" }, "Close", -1), He = /* @__PURE__ */ E({
  __name: "GlobalToastClose",
  emits: ["close"],
  setup(e, { emit: t }) {
    function n() {
      t("close");
    }
    return (s, o) => (h(), y("div", Nn, [
      d("button", {
        onClick: n,
        class: "rounded-md inline-flex text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      }, [
        Fn,
        w(p(I), {
          icon: "mdi-close-thick",
          class: "w-4 h-4"
        })
      ])
    ]));
  }
}), he = /* @__PURE__ */ E({
  __name: "GlobalToastProgress",
  props: { style: String },
  setup(e) {
    const t = e;
    return (n, s) => (h(), y("div", {
      class: "absolute left-0 bottom-0 right-0 h-1 rounded bg-gray-200 dark:bg-gray-600",
      style: X(t.style)
    }, null, 4));
  }
}), Dn = { class: "rounded-lg shadow-xs overflow-hidden z-100" }, Bn = { class: "p-4" }, qn = { class: "flex items-start" }, zn = {
  key: 0,
  class: "flex-shrink-0"
}, Hn = { class: "ml-3 w-0 flex-1 pt-0.5" }, Rn = {
  key: 0,
  class: "text-sm leading-5 font-medium"
}, Un = ["innerHTML"], Vn = { class: "ml-4 flex-shrink-0 flex mt-0.5" }, Gn = { class: "flex rounded-lg shadow-xs divide-x divide-gray-200 dark:divide-gray-600" }, Qn = { class: "w-0 flex-1 flex items-start p-4" }, Kn = {
  key: 0,
  class: "flex-shrink-0 mr-4"
}, Wn = { class: "w-full" }, Yn = {
  key: 0,
  class: "text-sm leading-5 font-medium text-gray-900 dark:text-gray-300"
}, Jn = ["innerHTML"], Xn = { class: "flex" }, Zn = { class: "flex flex-col divide-y divide-gray-200 dark:divide-gray-600" }, eo = { class: "h-0 flex-1 flex" }, to = { class: "h-0 flex-1 flex" }, no = { class: "rounded-lg shadow-xs overflow-hidden" }, oo = { class: "p-4" }, ro = { class: "flex items-center" }, so = {
  key: 0,
  class: "flex-shrink-0 mr-4"
}, io = { class: "w-0 flex-1 flex justify-between" }, ao = ["innerHTML"], lo = { class: "ml-4 flex-shrink-0 flex" }, z = /* @__PURE__ */ E({
  __name: "GlobalToast",
  props: {
    title: String,
    icon: {
      type: [Boolean, String],
      required: !1,
      default: void 0
    },
    message: {
      type: String
    },
    type: {
      type: String,
      required: !1,
      default: "info"
    },
    progress: {
      type: Boolean,
      required: !1,
      default: !0
    },
    timeout: {
      type: Number,
      required: !1,
      default: 2
    },
    pauseOnHover: {
      type: Boolean,
      required: !1,
      default: !0
    },
    primary: Object,
    secondary: Object,
    wide: Boolean
  },
  setup(e) {
    const t = e, n = A(void 0), s = A(!1), o = A(!1);
    let r;
    const i = A(0), a = 100, l = mt(), c = F(() => typeof t.icon == "string" ? t.icon : t.type), u = F(() => Math.round(i.value * 100 / (t.timeout * 1e3) * 100 / 100)), f = F(() => `width: ${u.value}%; transition: width 0.1s linear`);
    Ie(() => {
      s.value = !0, t.timeout > 0 && (i.value = t.timeout * 1e3, r = setInterval(() => g(), a));
    });
    function g() {
      (!t.pauseOnHover || !o.value) && (i.value -= a, i.value === 0 && b());
    }
    function b(q = !1) {
      clearInterval(r), q ? (s.value = !1, setTimeout(() => ke(n == null ? void 0 : n.value), 100), setTimeout(() => l == null ? void 0 : l.appContext.app.unmount(), 100)) : (setTimeout(() => s.value = !1, 100), setTimeout(() => ke(n == null ? void 0 : n.value), 200), setTimeout(() => l == null ? void 0 : l.appContext.app.unmount(), 200));
    }
    function x() {
      t.primary && t.primary.action(), b();
    }
    function L() {
      t.secondary && t.secondary.action(), b();
    }
    return (q, j) => (h(), y("div", {
      ref_key: "toastRef",
      ref: n,
      onMouseenter: j[2] || (j[2] = (O) => o.value = !0),
      onMouseleave: j[3] || (j[3] = (O) => o.value = !1)
    }, [
      w(me, {
        "enter-active-class": "transform ease-out duration-300 transition",
        "enter-from-class": "translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2",
        "enter-to-class": "translate-y-0 opacity-100 sm:translate-x-0",
        "leave-active-class": "transform ease-in duration-100 transition",
        "leave-from-class": "translate-y-0 opacity-100 sm:translate-y-0 sm:translate-x-0",
        "leave-to-class": "translate-y-1 opacity-0 sm:translate-y-0 sm:translate-x-1"
      }, {
        default: R(() => [
          s.value && !e.primary ? (h(), y("div", {
            key: 0,
            class: S([{ "max-w-sm": !e.wide, "max-w-3xl": e.wide }, "relative w-full bg-white dark:bg-gray-800 shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"])
          }, [
            e.progress && e.timeout ? (h(), C(he, {
              key: 0,
              style: X(p(f))
            }, null, 8, ["style"])) : v("", !0),
            d("div", Dn, [
              d("div", Bn, [
                d("div", qn, [
                  t.icon !== !1 ? (h(), y("div", zn, [
                    w(ge, { type: p(c) }, null, 8, ["type"])
                  ])) : v("", !0),
                  d("div", Hn, [
                    e.title ? (h(), y("p", Rn, _(e.title), 1)) : v("", !0),
                    d("p", {
                      class: S([{ "mt-1": e.title }, "text-sm leading-5"]),
                      innerHTML: e.message
                    }, null, 10, Un)
                  ]),
                  d("div", Vn, [
                    w(He, {
                      onClose: j[0] || (j[0] = (O) => b(!0))
                    })
                  ])
                ])
              ])
            ])
          ], 2)) : s.value && e.primary && e.secondary ? (h(), y("div", {
            key: 1,
            class: S([{ "max-w-sm": !e.wide, "max-w-3xl": e.wide }, "relative w-full bg-white dark:bg-gray-800 shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"])
          }, [
            e.progress && e.timeout ? (h(), C(he, {
              key: 0,
              style: X(p(f))
            }, null, 8, ["style"])) : v("", !0),
            d("div", Gn, [
              d("div", Qn, [
                t.icon !== !1 ? (h(), y("div", Kn, [
                  w(ge, { type: p(c) }, null, 8, ["type"])
                ])) : v("", !0),
                d("div", Wn, [
                  e.title ? (h(), y("p", Yn, _(e.title), 1)) : v("", !0),
                  d("p", {
                    class: S([{ "mt-1": e.title }, "text-sm leading-5 text-gray-500"]),
                    innerHTML: e.message
                  }, null, 10, Jn)
                ])
              ]),
              d("div", Xn, [
                d("div", Zn, [
                  d("div", eo, [
                    d("button", {
                      class: "w-full border border-transparent rounded-none rounded-tr-lg px-4 py-3 flex items-center justify-center text-sm font-medium text-gray-900 dark:text-gray-100 hover:text-gray-500 focus:outline-none focus:z-10 focus:ring-2 focus:ring-gray-500",
                      onClick: x
                    }, _(e.primary.label), 1)
                  ]),
                  d("div", to, [
                    d("button", {
                      class: "w-full border border-transparent rounded-none rounded-br-lg px-4 py-3 flex items-center justify-center text-sm font-medium text-gray-700 dark:text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-gray-500",
                      onClick: L
                    }, _(e.secondary.label), 1)
                  ])
                ])
              ])
            ])
          ], 2)) : s.value && e.primary && !e.secondary ? (h(), y("div", {
            key: 2,
            class: S([{ "max-w-sm": !e.wide, "max-w-3xl": e.wide }, "relative w-full bg-white dark:bg-gray-800 shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"])
          }, [
            e.progress && e.timeout ? (h(), C(he, {
              key: 0,
              style: X(p(f))
            }, null, 8, ["style"])) : v("", !0),
            d("div", no, [
              d("div", oo, [
                d("div", ro, [
                  t.icon !== !1 ? (h(), y("div", so, [
                    w(ge, { type: p(c) }, null, 8, ["type"])
                  ])) : v("", !0),
                  d("div", io, [
                    d("p", {
                      class: "w-0 flex-1 text-sm leading-5",
                      innerHTML: e.message
                    }, null, 8, ao),
                    d("button", {
                      type: "button",
                      class: "ml-3 flex-shrink-0 rounded-md text-sm font-medium text-gray-600 dark:text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-500",
                      onClick: x
                    }, _(e.primary.label), 1)
                  ]),
                  d("div", lo, [
                    d("button", {
                      class: "inline-flex text-gray-400 focus:outline-none focus:text-gray-500 transition ease-in-out duration-150",
                      onClick: b
                    }, [
                      w(He, {
                        onClose: j[1] || (j[1] = (O) => b(!0))
                      })
                    ])
                  ])
                ])
              ])
            ])
          ], 2)) : v("", !0)
        ]),
        _: 1
      })
    ], 544));
  }
}), co = [
  "z-40",
  "fixed",
  "inset-0",
  "flex",
  "sm:items-start",
  "px-4",
  "py-6",
  "pointer-events-none",
  "sm:p-6"
], uo = [
  "w-full",
  "flex",
  "flex-col-reverse",
  "items-center",
  "sm:items-end",
  "sm:justify-end"
], H = ["w-full", "max-w-sm", "mt-4"];
function ho() {
  const e = document.createElement("div");
  co.forEach((n) => e.classList.add(n)), e.setAttribute("aria-alive", "assertive");
  const t = document.createElement("div");
  return uo.forEach((n) => t.classList.add(n)), t.setAttribute("id", "toasts"), e.appendChild(t), document.body.appendChild(e), {
    show(n) {
      if (typeof n == "string" && (n = { message: n }), n !== void 0)
        return N("toasts", n, z, H);
    },
    success(n) {
      return N("toasts", { type: "success", message: n }, z, H);
    },
    info(n) {
      return N("toasts", { type: "info", message: n }, z, H);
    },
    danger(n) {
      return N("toasts", { type: "danger", message: n, timeout: 20 }, z, H);
    },
    warning(n) {
      return N("toasts", { type: "warning", message: n, timeout: 10 }, z, H);
    },
    denied(n) {
      return N("toasts", { type: "denied", message: n, timeout: 10 }, z, H);
    }
  };
}
const mo = (e) => {
  e.component("PushButton", ye), e.component("ModalBase", lt);
};
export {
  lt as ModalBase,
  ye as PushButton,
  mo as install,
  go as useModal,
  ho as useToast
};
