// @inkline/inkline/components/IHamburgerMenu/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "IHamburgerMenu";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  inheritAttrs: false,
  props: {
    /**
     * The animation of the hamburger menu
     * @type close | arrow-up | arrow-down | arrow-left | arrow-right | plus | minus
     * @default close
     * @name animation
     */
    animation: {
      type: String,
      default: "close"
    },
    /**
     * The color variant of the hamburger menu
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "color")
    },
    /**
     * Used to set the hamburger menu as opened or closed
     * @type Boolean
     * @default false
     * @name modelValue
     */
    modelValue: {
      type: Boolean,
      default: false
    }
  },
  emits: [
  /**
   * Event emitted for setting the modelValue
   * @event update:modelValue
   */
  "update:modelValue"],
  computed: {
    classes() {
      return {
        ...(0, _mixins.colorVariantClass)(this),
        "-active": this.modelValue,
        [`-${this.animation}`]: true
      };
    }
  },
  methods: {
    onClick() {
      this.$emit("update:modelValue", !this.modelValue);
    }
  }
});
module.exports = _default;
