// @inkline/inkline/components/IToggle/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IToggleBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IToggleColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "IToggleDisabledExample", {
  enumerable: true,
  get: function () {
    return _disabled.default;
  }
});
Object.defineProperty(exports, "IToggleReadonlyExample", {
  enumerable: true,
  get: function () {
    return _readonly.default;
  }
});
Object.defineProperty(exports, "IToggleRoundedExample", {
  enumerable: true,
  get: function () {
    return _rounded.default;
  }
});
Object.defineProperty(exports, "IToggleSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/basic.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/color-variants.vue"));
var _disabled = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/disabled.vue"));
var _readonly = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/readonly.vue"));
var _rounded = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/rounded.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/IToggle/examples/size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
