// nuxt-bugsnag/dist/module.mjs
import { createResolver, defineNuxtModule, isNuxt3, addPlugin, extendViteConfig } from '@nuxt/kit';
import { node, browser } from '@bugsnag/source-maps';

const { resolve } = createResolver(import.meta.url);
const module = defineNuxtModule({
  meta: {
    name: "nuxt-bugsnag",
    configKey: "bugsnag",
    compatibility: {
      nuxt: "^3.0.0-rc.10 || ^3.0.0 || ^2.16.0",
      bridge: true
    }
  },
  defaults: {
    disabled: false,
    publishRelease: false,
    baseUrl: "http://localhost:3000",
    config: {
      notifyReleaseStages: [],
      apiKey: "",
      environment: "production",
      appVersion: "1.0.0"
    },
    projectRoot: "/"
  },
  hooks: {
    "imports:extend": (imports) => {
      imports.push({
        name: "useBugsnag",
        as: "useBugsnag",
        from: resolve("./runtime/composables/useBugsnag")
      });
    }
  },
  setup(options, nuxt) {
    if (options.disabled) {
      return;
    }
    if (isNuxt3()) {
      nuxt.options.runtimeConfig.public.bugsnag = options.config;
    } else {
      nuxt.options.publicRuntimeConfig.bugsnag = options.config;
    }
    addPlugin(resolve("./runtime/plugin"));
    extendViteConfig((config) => {
      config.optimizeDeps?.include?.push(
        ...["@bugsnag/plugin-vue", "@bugsnag/js"]
      );
    });
    if (!options.publishRelease || nuxt.options.dev) {
      return;
    }
    nuxt.options.sourcemap = { server: true, client: true };
    nuxt.addHooks({
      "nitro:init": (nitro) => {
        nitro.hooks.addHooks({
          compiled: async (nitro2) => {
            nitro2.logger.log("");
            nitro2.logger.start("upload of sourcemaps to bugsnag \n");
            const promises = [];
            promises.push(
              node.uploadMultiple({
                apiKey: options.config.apiKey,
                appVersion: options.config.appVersion,
                directory: nitro2.options.output.serverDir,
                logger: nitro2.logger,
                overwrite: true,
                projectRoot: options.projectRoot
              })
            );
            promises.push(
              browser.uploadMultiple({
                apiKey: options.config.apiKey,
                appVersion: options.config.appVersion,
                directory: nitro2.options.output.publicDir,
                logger: nitro2.logger,
                overwrite: true,
                baseUrl: options.baseUrl
              })
            );
            await Promise.all(promises);
            nitro2.logger.log("");
            nitro2.logger.success("upload of sourcemaps to bugsnag \n");
          }
        });
      }
    });
  }
});

export { module as default };
