// @inkline/inkline/components/IForm/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IFormBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IFormDisabledExample", {
  enumerable: true,
  get: function () {
    return _disabled.default;
  }
});
Object.defineProperty(exports, "IFormReadonlyExample", {
  enumerable: true,
  get: function () {
    return _readonly.default;
  }
});
Object.defineProperty(exports, "IFormSizeVariantsLgExample", {
  enumerable: true,
  get: function () {
    return _sizeVariantsLg.default;
  }
});
Object.defineProperty(exports, "IFormSizeVariantsMdExample", {
  enumerable: true,
  get: function () {
    return _sizeVariantsMd.default;
  }
});
Object.defineProperty(exports, "IFormSizeVariantsSmExample", {
  enumerable: true,
  get: function () {
    return _sizeVariantsSm.default;
  }
});
var _basic = _interopRequireDefault(require("./basic.vue"));
var _disabled = _interopRequireDefault(require("./disabled.vue"));
var _readonly = _interopRequireDefault(require("./readonly.vue"));
var _sizeVariantsSm = _interopRequireDefault(require("./size-variants-sm.vue"));
var _sizeVariantsMd = _interopRequireDefault(require("./size-variants-md.vue"));
var _sizeVariantsLg = _interopRequireDefault(require("./size-variants-lg.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
