// @inkline/inkline/components/IButton/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IButtonBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IButtonBlockExample", {
  enumerable: true,
  get: function () {
    return _block.default;
  }
});
Object.defineProperty(exports, "IButtonCircleExample", {
  enumerable: true,
  get: function () {
    return _circle.default;
  }
});
Object.defineProperty(exports, "IButtonColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "IButtonIconExample", {
  enumerable: true,
  get: function () {
    return _icon.default;
  }
});
Object.defineProperty(exports, "IButtonLinkExample", {
  enumerable: true,
  get: function () {
    return _link.default;
  }
});
Object.defineProperty(exports, "IButtonOutlineExample", {
  enumerable: true,
  get: function () {
    return _outline.default;
  }
});
Object.defineProperty(exports, "IButtonRoutingExample", {
  enumerable: true,
  get: function () {
    return _routing.default;
  }
});
Object.defineProperty(exports, "IButtonSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
Object.defineProperty(exports, "IButtonStateActiveExample", {
  enumerable: true,
  get: function () {
    return _stateActive.default;
  }
});
Object.defineProperty(exports, "IButtonStateDisabledExample", {
  enumerable: true,
  get: function () {
    return _stateDisabled.default;
  }
});
Object.defineProperty(exports, "IButtonStateLoadingExample", {
  enumerable: true,
  get: function () {
    return _stateLoading.default;
  }
});
Object.defineProperty(exports, "IButtonTagExample", {
  enumerable: true,
  get: function () {
    return _buttonTag.default;
  }
});
Object.defineProperty(exports, "IButtonTypeExample", {
  enumerable: true,
  get: function () {
    return _buttonType.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/basic.vue"));
var _block = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/block.vue"));
var _buttonTag = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/button-tag.vue"));
var _buttonType = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/button-type.vue"));
var _circle = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/circle.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/color-variants.vue"));
var _icon = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/icon.vue"));
var _link = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/link.vue"));
var _outline = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/outline.vue"));
var _routing = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/routing.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/size-variants.vue"));
var _stateActive = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/state-active.vue"));
var _stateDisabled = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/state-disabled.vue"));
var _stateLoading = _interopRequireDefault(require("@inkline/inkline/components/IButton/examples/state-loading.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
