// @inkline/inkline/components/IListGroup/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _mixins = require("@inkline/inkline/mixins");
const componentName = "IListGroup";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  inheritAttrs: false,
  props: {
    /**
     * Display the list group border
     * @type Boolean
     * @default true
     * @name border
     */
    border: {
      type: Boolean,
      default: true
    },
    /**
     * The color variant of the list group
     * @type light | dark
     * @default light
     * @name color
     */
    color: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "color")
    },
    /**
     * The size variant of the list group
     * @type sm | md | lg
     * @default md
     * @name size
     */
    size: {
      type: String,
      default: (0, _mixins.defaultPropValue)(componentName, "size"),
      validator: _mixins.sizePropValidator
    }
  },
  computed: {
    classes() {
      return {
        ...(0, _mixins.colorVariantClass)(this),
        [`-${this.size}`]: Boolean(this.size),
        "-border": this.border
      };
    }
  }
});
module.exports = _default;
