// @inkline/inkline/components/ILayout/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ILayoutBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "ILayoutContentHeaderExample", {
  enumerable: true,
  get: function () {
    return _contentHeader.default;
  }
});
Object.defineProperty(exports, "ILayoutContentHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _contentHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutContentWithLeftAndRightAsidesHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _contentWithLeftAndRightAsidesHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutContentWithLeftAsideHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _contentWithLeftAsideHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutContentWithRightAsideHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _contentWithRightAsideHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutLeftAndRightAsidesWithContentHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _leftAndRightAsidesWithContentHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutLeftAsideWithContentHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _leftAsideWithContentHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutRightAsideWithContentHeaderFooterExample", {
  enumerable: true,
  get: function () {
    return _rightAsideWithContentHeaderFooter.default;
  }
});
Object.defineProperty(exports, "ILayoutWithAsideCSSExample", {
  enumerable: true,
  get: function () {
    return _withAside.default;
  }
});
var _basic = _interopRequireDefault(require("./basic.vue"));
var _contentHeader = _interopRequireDefault(require("./content-header.vue"));
var _contentHeaderFooter = _interopRequireDefault(require("./content-header-footer.vue"));
var _contentWithLeftAsideHeaderFooter = _interopRequireDefault(require("./content-with-left-aside-header-footer.vue"));
var _contentWithLeftAndRightAsidesHeaderFooter = _interopRequireDefault(require("./content-with-left-and-right-asides-header-footer.vue"));
var _contentWithRightAsideHeaderFooter = _interopRequireDefault(require("./content-with-right-aside-header-footer.vue"));
var _leftAndRightAsidesWithContentHeaderFooter = _interopRequireDefault(require("./left-and-right-asides-with-content-header-footer.vue"));
var _leftAsideWithContentHeaderFooter = _interopRequireDefault(require("./left-aside-with-content-header-footer.vue"));
var _rightAsideWithContentHeaderFooter = _interopRequireDefault(require("./right-aside-with-content-header-footer.vue"));
var _withAside = _interopRequireDefault(require("./with-aside.scss"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
