// @inkline/inkline/components/IAlert/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IAlertBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "IAlertColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "IAlertContentExample", {
  enumerable: true,
  get: function () {
    return _content.default;
  }
});
Object.defineProperty(exports, "IAlertDismissibleExample", {
  enumerable: true,
  get: function () {
    return _dismissible.default;
  }
});
Object.defineProperty(exports, "IAlertIconExample", {
  enumerable: true,
  get: function () {
    return _icon.default;
  }
});
Object.defineProperty(exports, "IAlertSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/basic.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/color-variants.vue"));
var _dismissible = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/dismissible.vue"));
var _content = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/content.vue"));
var _icon = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/icon.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/IAlert/examples/size-variants.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
