// @inkline/inkline/components/INav/examples/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "INavBasicExample", {
  enumerable: true,
  get: function () {
    return _basic.default;
  }
});
Object.defineProperty(exports, "INavColorVariantsExample", {
  enumerable: true,
  get: function () {
    return _colorVariants.default;
  }
});
Object.defineProperty(exports, "INavRoutingExample", {
  enumerable: true,
  get: function () {
    return _routing.default;
  }
});
Object.defineProperty(exports, "INavSizeVariantsExample", {
  enumerable: true,
  get: function () {
    return _sizeVariants.default;
  }
});
Object.defineProperty(exports, "INavStateActiveExample", {
  enumerable: true,
  get: function () {
    return _stateActive.default;
  }
});
Object.defineProperty(exports, "INavVerticalExample", {
  enumerable: true,
  get: function () {
    return _vertical.default;
  }
});
var _basic = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/basic.vue"));
var _colorVariants = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/color-variants.vue"));
var _routing = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/routing.vue"));
var _sizeVariants = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/size-variants.vue"));
var _stateActive = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/state-active.vue"));
var _vertical = _interopRequireDefault(require("@inkline/inkline/components/INav/examples/vertical.vue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
