// @inkline/inkline/components/IRow/script.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _vue = require("vue");
var _constants = require("@inkline/inkline/constants");
var _helpers = require("@inkline/inkline/helpers");
const properties = {};
for (const breakpoint of _constants.breakpointKeys) {
  for (const property of ["start", "center", "end", "top", "middle", "bottom", "around", "between", "reverse"]) {
    properties[`${property}${(0, _helpers.capitalizeFirst)(breakpoint)}`] = {
      type: Boolean,
      default: false
    };
  }
}
const componentName = "IRow";
var _default = (0, _vue.defineComponent)({
  name: componentName,
  inheritAttrs: false,
  props: {
    noGutter: {
      type: Boolean,
      default: false
    },
    noCollapse: {
      type: Boolean,
      default: false
    },
    ...properties
  },
  computed: {
    classes() {
      const responsiveClasses = Object.keys(properties).reduce((acc, property) => {
        if (this[property]) {
          acc[(0, _helpers.breakpointClass)(`-${property}`, this[property])] = true;
        }
        return acc;
      }, {});
      return {
        "-no-gutter": this.noGutter,
        "-no-collapse": this.noCollapse,
        ...responsiveClasses
      };
    }
  }
});
module.exports = _default;
