// nuxt-viewport/dist/module.mjs
import { fileURLToPath } from 'url';
import { defineNuxtModule, createResolver, addTemplate, addPlugin, addImports } from '@nuxt/kit';
import 'vue-demi';

const name = "nuxt-viewport";
const version = "2.0.4";

const DEFAULT_OPTIONS = {
  breakpoints: {
    desktop: 1024,
    desktopMedium: 1280,
    desktopWide: 1600,
    mobile: 320,
    mobileMedium: 375,
    mobileWide: 425,
    tablet: 768
  },
  cookieName: "viewport",
  defaultBreakpoints: {
    desktop: "desktop",
    mobile: "mobile",
    tablet: "tablet"
  },
  fallbackBreakpoint: "desktop"
};

const module = defineNuxtModule({
  meta: {
    configKey: "viewport",
    compatibility: {
      bridge: true
    },
    name,
    version
  },
  setup(options, nuxt) {
    options = {
      ...DEFAULT_OPTIONS,
      ...options
    };
    const { resolve } = createResolver(import.meta.url);
    const runtimeDir = fileURLToPath(new URL("./runtime", import.meta.url));
    nuxt.options.build.transpile.push(runtimeDir);
    nuxt.options.alias["#viewport-options"] = addTemplate({
      filename: "viewport-options.mjs",
      getContents() {
        return `export default ${JSON.stringify(options)}`;
      }
    }).dst;
    addPlugin(resolve(runtimeDir, "plugin"));
    addImports({
      as: "useViewport",
      from: resolve(runtimeDir, "composables"),
      name: "useViewport"
    });
  }
});

export { module as default };
